// cdk-iam-floyd/lib/shared/policy-statement/7-principals-CDK.js
"use strict";
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
exports.PolicyStatementWithCDKPrincipal = void 0;
const JSII_RTTI_SYMBOL_1 = Symbol.for("jsii.rtti");
// @ts-ignore only available after swapping 1-base
const aws_cdk_lib_1 = require("aws-cdk-lib");
const _6_principals_1 = require("./6-principals");
/**
 * (experimental) Adds CDK specific "principal" functionality to the Policy Statement.
 *
 * @experimental
 */
class PolicyStatementWithCDKPrincipal extends _6_principals_1.PolicyStatementWithPrincipal {
    constructor() {
        super(...arguments);
        /**
         * @experimental
         */
        this.cdkPrincipals = [];
        this.cdkPrincipalsApplied = false;
    }
    /**
     * (experimental) Adds any number of CDK principals to the statement.
     *
     * @param principals Principals to add to the statement.
     * @experimental
     */
    forCdkPrincipal(...principals) {
        this.skipAutoResource = true;
        this.cdkPrincipals.push(...principals);
        return this;
    }
    /**
     * @experimental
     */
    cdkApplyPrincipals() {
        if (!this.cdkPrincipalsApplied) {
            const mode = this.useNotPrincipals ? 'addNotPrincipals' : 'addPrincipals';
            // @ts-ignore only available after swapping 1-base
            this[mode](...this.cdkPrincipals);
            if (this.hasPrincipals()) {
                Object.keys(this.principals).forEach((prefix) => {
                    switch (prefix) {
                        case _6_principals_1.PrincipalType.AWS:
                            this.principals[prefix].forEach((arn) => {
                                // @ts-ignore only available after swapping 1-base
                                this[mode](new aws_cdk_lib_1.aws_iam.ArnPrincipal(arn));
                            });
                            break;
                        case _6_principals_1.PrincipalType.CANONICAL_USER:
                            this.principals[prefix].forEach((userId) => {
                                // @ts-ignore only available after swapping 1-base
                                this[mode](new aws_cdk_lib_1.aws_iam.CanonicalUserPrincipal(userId));
                            });
                            break;
                        case _6_principals_1.PrincipalType.FEDERATED:
                            this.principals[prefix].forEach((provider) => {
                                // @ts-ignore only available after swapping 1-base
                                this[mode](new aws_cdk_lib_1.aws_iam.FederatedPrincipal(provider, {}));
                            });
                            break;
                        case _6_principals_1.PrincipalType.SERVICE:
                            this.principals[prefix].forEach((service) => {
                                // @ts-ignore only available after swapping 1-base
                                this[mode](new aws_cdk_lib_1.aws_iam.ServicePrincipal(service));
                            });
                            break;
                        default:
                            throw Error(`Unhandled principal type: ${prefix}`);
                    }
                });
            }
            this.cdkPrincipalsApplied = true;
        }
    }
}
exports.PolicyStatementWithCDKPrincipal = PolicyStatementWithCDKPrincipal;
_a = JSII_RTTI_SYMBOL_1;
PolicyStatementWithCDKPrincipal[_a] = { fqn: "cdk-iam-floyd.PolicyStatementWithCDKPrincipal", version: "0.300.0" };
