// cdk-iam-floyd/lib/generated/applicationcostprofilerservice.js
"use strict";
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApplicationCostProfiler = void 0;
const JSII_RTTI_SYMBOL_1 = Symbol.for("jsii.rtti");
const shared_1 = require("../shared");
/**
 * (experimental) Statement provider for service [application-cost-profiler](https://docs.aws.amazon.com/service-authorization/latest/reference/list_awsapplicationcostprofilerservice.html).
 *
 * @experimental
 */
class ApplicationCostProfiler extends shared_1.PolicyStatement {
    /**
     * (experimental) Statement provider for service [application-cost-profiler](https://docs.aws.amazon.com/service-authorization/latest/reference/list_awsapplicationcostprofilerservice.html).
     *
     * @experimental
     */
    constructor(props) {
        super(props);
        /**
         * (experimental) Holds the prefix of the service actions, e.g. `ec2`.
         *
         * @experimental
         */
        this.servicePrefix = 'application-cost-profiler';
        /**
         * @experimental
         */
        this.accessLevelList = {
            Write: [
                'DeleteReportDefinition',
                'ImportApplicationUsage',
                'PutReportDefinition',
                'UpdateReportDefinition'
            ],
            Read: [
                'GetReportDefinition',
                'ListReportDefinitions'
            ]
        };
    }
    /**
     * (experimental) Grants permission to delete the configuration with specific Application Cost Profiler Report thereby effectively disabling report generation.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/application-cost-profiler/latest/APIReference/API_DeleteReportDefinition.html
     *
     * @experimental
     */
    toDeleteReportDefinition() {
        return this.to('DeleteReportDefinition');
    }
    /**
     * (experimental) Grants permission to fetch the configuration with specific Application Cost Profiler Report request.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/application-cost-profiler/latest/APIReference/API_GetReportDefinition.html
     *
     * @experimental
     */
    toGetReportDefinition() {
        return this.to('GetReportDefinition');
    }
    /**
     * (experimental) Grants permission to import the application usage from S3.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/application-cost-profiler/latest/APIReference/API_ImportApplicationUsage.html
     *
     * @experimental
     */
    toImportApplicationUsage() {
        return this.to('ImportApplicationUsage');
    }
    /**
     * (experimental) Grants permission to get a list of the different Application Cost Profiler Report configurations they have created.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/application-cost-profiler/latest/APIReference/API_ListReportDefinitions.html
     *
     * @experimental
     */
    toListReportDefinitions() {
        return this.to('ListReportDefinitions');
    }
    /**
     * (experimental) Grants permission to create Application Cost Profiler Report configurations.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/application-cost-profiler/latest/APIReference/API_PutReportDefinition.html
     *
     * @experimental
     */
    toPutReportDefinition() {
        return this.to('PutReportDefinition');
    }
    /**
     * (experimental) Grants permission to update an existing Application Cost Profiler Report configuration.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/application-cost-profiler/latest/APIReference/API_UpdateReportDefinition.html
     *
     * @experimental
     */
    toUpdateReportDefinition() {
        return this.to('UpdateReportDefinition');
    }
}
exports.ApplicationCostProfiler = ApplicationCostProfiler;
_a = JSII_RTTI_SYMBOL_1;
ApplicationCostProfiler[_a] = { fqn: "cdk-iam-floyd.ApplicationCostProfiler", version: "0.300.0" };
