// cdk-iam-floyd/lib/generated/cloudcontrolapi.js
"use strict";
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
exports.AwsCloudControlApi = void 0;
const JSII_RTTI_SYMBOL_1 = Symbol.for("jsii.rtti");
const shared_1 = require("../shared");
/**
 * (experimental) Statement provider for service [aws-cloud-control-api](https://docs.aws.amazon.com/service-authorization/latest/reference/list_awscloudcontrolapi.html).
 *
 * @experimental
 */
class AwsCloudControlApi extends shared_1.PolicyStatement {
    /**
     * (experimental) Statement provider for service [aws-cloud-control-api](https://docs.aws.amazon.com/service-authorization/latest/reference/list_awscloudcontrolapi.html).
     *
     * @experimental
     */
    constructor(props) {
        super(props);
        /**
         * (experimental) Holds the prefix of the service actions, e.g. `ec2`.
         *
         * @experimental
         */
        this.servicePrefix = 'cloudformation';
        /**
         * @experimental
         */
        this.accessLevelList = {
            Write: [
                'CancelResourceRequest',
                'CreateResource',
                'DeleteResource',
                'UpdateResource'
            ],
            Read: [
                'GetResource',
                'GetResourceRequestStatus',
                'ListResourceRequests',
                'ListResources'
            ]
        };
    }
    /**
     * (experimental) Grants permission to cancel resource requests in your account.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/cloudcontrolapi/latest/APIReference/API_CancelResourceRequest.html
     *
     * @experimental
     */
    toCancelResourceRequest() {
        return this.to('CancelResourceRequest');
    }
    /**
     * (experimental) Grants permission to create resources in your account.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/cloudcontrolapi/latest/APIReference/API_CreateResource.html
     *
     * @experimental
     */
    toCreateResource() {
        return this.to('CreateResource');
    }
    /**
     * (experimental) Grants permission to delete resources in your account.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/cloudcontrolapi/latest/APIReference/API_DeleteResource.html
     *
     * @experimental
     */
    toDeleteResource() {
        return this.to('DeleteResource');
    }
    /**
     * (experimental) Grants permission to get resources in your account.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/cloudcontrolapi/latest/APIReference/API_GetResource.html
     *
     * @experimental
     */
    toGetResource() {
        return this.to('GetResource');
    }
    /**
     * (experimental) Grants permission to get resource requests in your account.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/cloudcontrolapi/latest/APIReference/API_GetResourceRequestStatus.html
     *
     * @experimental
     */
    toGetResourceRequestStatus() {
        return this.to('GetResourceRequestStatus');
    }
    /**
     * (experimental) Grants permission to list resource requests in your account.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/cloudcontrolapi/latest/APIReference/API_ListResourceRequests.html
     *
     * @experimental
     */
    toListResourceRequests() {
        return this.to('ListResourceRequests');
    }
    /**
     * (experimental) Grants permission to list resources in your account.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/cloudcontrolapi/latest/APIReference/API_ListResources.html
     *
     * @experimental
     */
    toListResources() {
        return this.to('ListResources');
    }
    /**
     * (experimental) Grants permission to update resources in your account.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/cloudcontrolapi/latest/APIReference/API_UpdateResource.html
     *
     * @experimental
     */
    toUpdateResource() {
        return this.to('UpdateResource');
    }
}
exports.AwsCloudControlApi = AwsCloudControlApi;
_a = JSII_RTTI_SYMBOL_1;
AwsCloudControlApi[_a] = { fqn: "cdk-iam-floyd.AwsCloudControlApi", version: "0.300.0" };
