// cdk-iam-floyd/lib/generated/comprehendmedical.js
"use strict";
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
exports.Comprehendmedical = void 0;
const JSII_RTTI_SYMBOL_1 = Symbol.for("jsii.rtti");
const shared_1 = require("../shared");
/**
 * (experimental) Statement provider for service [comprehendmedical](https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazoncomprehendmedical.html).
 *
 * @experimental
 */
class Comprehendmedical extends shared_1.PolicyStatement {
    /**
     * (experimental) Statement provider for service [comprehendmedical](https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazoncomprehendmedical.html).
     *
     * @experimental
     */
    constructor(props) {
        super(props);
        /**
         * (experimental) Holds the prefix of the service actions, e.g. `ec2`.
         *
         * @experimental
         */
        this.servicePrefix = 'comprehendmedical';
        /**
         * @experimental
         */
        this.accessLevelList = {
            Read: [
                'DescribeEntitiesDetectionV2Job',
                'DescribeICD10CMInferenceJob',
                'DescribePHIDetectionJob',
                'DescribeRxNormInferenceJob',
                'DetectEntitiesV2',
                'DetectPHI',
                'InferICD10CM',
                'InferRxNorm',
                'ListEntitiesDetectionV2Jobs',
                'ListICD10CMInferenceJobs',
                'ListPHIDetectionJobs',
                'ListRxNormInferenceJobs'
            ],
            Write: [
                'StartEntitiesDetectionV2Job',
                'StartICD10CMInferenceJob',
                'StartPHIDetectionJob',
                'StartRxNormInferenceJob',
                'StopEntitiesDetectionV2Job',
                'StopICD10CMInferenceJob',
                'StopPHIDetectionJob',
                'StopRxNormInferenceJob'
            ]
        };
    }
    /**
     * (experimental) Grants permission to describe the properties of a medical entity detection job that you have submitted.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/comprehend/latest/dg/API_medical_DescribeEntitiesDetectionV2Job.html
     *
     * @experimental
     */
    toDescribeEntitiesDetectionV2Job() {
        return this.to('DescribeEntitiesDetectionV2Job');
    }
    /**
     * (experimental) Grants permission to describe the properties of an ICD-10-CM linking job that you have submitted.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/comprehend/latest/dg/API_medical_DescribeICD10CMInferenceJob.html
     *
     * @experimental
     */
    toDescribeICD10CMInferenceJob() {
        return this.to('DescribeICD10CMInferenceJob');
    }
    /**
     * (experimental) Grants permission to describe the properties of a PHI entity detection job that you have submitted.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/comprehend/latest/dg/API_medical_DescribePHIDetectionJob.html
     *
     * @experimental
     */
    toDescribePHIDetectionJob() {
        return this.to('DescribePHIDetectionJob');
    }
    /**
     * (experimental) Grants permission to describe the properties of an RxNorm linking job that you have submitted.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/comprehend/latest/dg/API_medical_DescribeRxNormInferenceJob.html
     *
     * @experimental
     */
    toDescribeRxNormInferenceJob() {
        return this.to('DescribeRxNormInferenceJob');
    }
    /**
     * (experimental) Grants permission to detect the named medical entities, and their relationships and traits within the given text document.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/comprehend/latest/dg/API_medical_DetectEntitiesV2.html
     *
     * @experimental
     */
    toDetectEntitiesV2() {
        return this.to('DetectEntitiesV2');
    }
    /**
     * (experimental) Grants permission to detect the protected health information (PHI) entities within the given text document.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/comprehend/latest/dg/API_medical_DetectPHI.html
     *
     * @experimental
     */
    toDetectPHI() {
        return this.to('DetectPHI');
    }
    /**
     * (experimental) Grants permission to detect the medical condition entities within the given text document and link them to ICD-10-CM codes.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/comprehend/latest/dg/API_medical_InferICD10CM.html
     *
     * @experimental
     */
    toInferICD10CM() {
        return this.to('InferICD10CM');
    }
    /**
     * (experimental) Grants permission to detect the medication entities within the given text document and link them to RxCUI concept identifiers from the National Library of Medicine RxNorm database.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/comprehend/latest/dg/API_medical_InferRxNorm.html
     *
     * @experimental
     */
    toInferRxNorm() {
        return this.to('InferRxNorm');
    }
    /**
     * (experimental) Grants permission to list the medical entity detection jobs that you have submitted.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/comprehend/latest/dg/API_medical_ListEntitiesDetectionV2Jobs.html
     *
     * @experimental
     */
    toListEntitiesDetectionV2Jobs() {
        return this.to('ListEntitiesDetectionV2Jobs');
    }
    /**
     * (experimental) Grants permission to list the ICD-10-CM linking jobs that you have submitted.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/comprehend/latest/dg/API_medical_ListICD10CMInferenceJobs.html
     *
     * @experimental
     */
    toListICD10CMInferenceJobs() {
        return this.to('ListICD10CMInferenceJobs');
    }
    /**
     * (experimental) Grants permission to list the PHI entity detection jobs that you have submitted.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/comprehend/latest/dg/API_medical_ListPHIDetectionJobs.html
     *
     * @experimental
     */
    toListPHIDetectionJobs() {
        return this.to('ListPHIDetectionJobs');
    }
    /**
     * (experimental) Grants permission to list the RxNorm linking jobs that you have submitted.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/comprehend/latest/dg/API_medical_ListRxNormInferenceJobs.html
     *
     * @experimental
     */
    toListRxNormInferenceJobs() {
        return this.to('ListRxNormInferenceJobs');
    }
    /**
     * (experimental) Grants permission to start an asynchronous medical entity detection job for a collection of documents.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/comprehend/latest/dg/API_medical_StartEntitiesDetectionV2Job.html
     *
     * @experimental
     */
    toStartEntitiesDetectionV2Job() {
        return this.to('StartEntitiesDetectionV2Job');
    }
    /**
     * (experimental) Grants permission to start an asynchronous ICD-10-CM linking job for a collection of documents.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/comprehend/latest/dg/API_medical_StartICD10CMInferenceJob.html
     *
     * @experimental
     */
    toStartICD10CMInferenceJob() {
        return this.to('StartICD10CMInferenceJob');
    }
    /**
     * (experimental) Grants permission to start an asynchronous PHI entity detection job for a collection of documents.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/comprehend/latest/dg/API_medical_StartPHIDetectionJob.html
     *
     * @experimental
     */
    toStartPHIDetectionJob() {
        return this.to('StartPHIDetectionJob');
    }
    /**
     * (experimental) Grants permission to start an asynchronous RxNorm linking job for a collection of documents.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/comprehend/latest/dg/API_medical_StartRxNormInferenceJob.html
     *
     * @experimental
     */
    toStartRxNormInferenceJob() {
        return this.to('StartRxNormInferenceJob');
    }
    /**
     * (experimental) Grants permission to stop a medical entity detection job.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/comprehend/latest/dg/API_medical_StopEntitiesDetectionV2Job.html
     *
     * @experimental
     */
    toStopEntitiesDetectionV2Job() {
        return this.to('StopEntitiesDetectionV2Job');
    }
    /**
     * (experimental) Grants permission to stop an ICD-10-CM linking job.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/comprehend/latest/dg/API_medical_StopICD10CMInferenceJob.html
     *
     * @experimental
     */
    toStopICD10CMInferenceJob() {
        return this.to('StopICD10CMInferenceJob');
    }
    /**
     * (experimental) Grants permission to stop a PHI entity detection job.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/comprehend/latest/dg/API_medical_StopPHIDetectionJob.html
     *
     * @experimental
     */
    toStopPHIDetectionJob() {
        return this.to('StopPHIDetectionJob');
    }
    /**
     * (experimental) Grants permission to stop an RxNorm linking job.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/comprehend/latest/dg/API_medical_StopRxNormInferenceJob.html
     *
     * @experimental
     */
    toStopRxNormInferenceJob() {
        return this.to('StopRxNormInferenceJob');
    }
}
exports.Comprehendmedical = Comprehendmedical;
_a = JSII_RTTI_SYMBOL_1;
Comprehendmedical[_a] = { fqn: "cdk-iam-floyd.Comprehendmedical", version: "0.300.0" };
