// cdk-iam-floyd/lib/generated/connectwisdom.js
"use strict";
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
exports.Wisdom = void 0;
const JSII_RTTI_SYMBOL_1 = Symbol.for("jsii.rtti");
const shared_1 = require("../shared");
/**
 * (experimental) Statement provider for service [wisdom](https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazonconnectwisdom.html).
 *
 * @experimental
 */
class Wisdom extends shared_1.PolicyStatement {
    /**
     * (experimental) Statement provider for service [wisdom](https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazonconnectwisdom.html).
     *
     * @experimental
     */
    constructor(props) {
        super(props);
        /**
         * (experimental) Holds the prefix of the service actions, e.g. `ec2`.
         *
         * @experimental
         */
        this.servicePrefix = 'wisdom';
        /**
         * @experimental
         */
        this.accessLevelList = {
            Write: [
                'CreateAssistant',
                'CreateAssistantAssociation',
                'CreateContent',
                'CreateKnowledgeBase',
                'CreateSession',
                'DeleteAssistant',
                'DeleteAssistantAssociation',
                'DeleteContent',
                'DeleteKnowledgeBase',
                'NotifyRecommendationsReceived',
                'RemoveKnowledgeBaseTemplateUri',
                'StartContentUpload',
                'UpdateContent',
                'UpdateKnowledgeBaseTemplateUri'
            ],
            Read: [
                'GetAssistant',
                'GetAssistantAssociation',
                'GetContent',
                'GetContentSummary',
                'GetKnowledgeBase',
                'GetRecommendations',
                'GetSession',
                'ListTagsForResource',
                'QueryAssistant',
                'SearchContent',
                'SearchSessions'
            ],
            List: [
                'ListAssistantAssociations',
                'ListAssistants',
                'ListContents',
                'ListKnowledgeBases'
            ],
            Tagging: [
                'TagResource',
                'UntagResource'
            ]
        };
    }
    /**
     * (experimental) Grants permission to create an assistant.
     *
     * Access Level: Write
     *
     * Possible conditions:
     * - .ifAwsTagKeys()
     * - .ifAwsRequestTag()
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_CreateAssistant.html
     *
     * @experimental
     */
    toCreateAssistant() {
        return this.to('CreateAssistant');
    }
    /**
     * (experimental) Grants permission to create an association between an assistant and another resource.
     *
     * Access Level: Write
     *
     * Possible conditions:
     * - .ifAwsTagKeys()
     * - .ifAwsRequestTag()
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_CreateAssistantAssociation.html
     *
     * @experimental
     */
    toCreateAssistantAssociation() {
        return this.to('CreateAssistantAssociation');
    }
    /**
     * (experimental) Grants permission to create content.
     *
     * Access Level: Write
     *
     * Possible conditions:
     * - .ifAwsTagKeys()
     * - .ifAwsRequestTag()
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_CreateContent.html
     *
     * @experimental
     */
    toCreateContent() {
        return this.to('CreateContent');
    }
    /**
     * (experimental) Grants permission to create a knowledge base.
     *
     * Access Level: Write
     *
     * Possible conditions:
     * - .ifAwsTagKeys()
     * - .ifAwsRequestTag()
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_CreateKnowledgeBase.html
     *
     * @experimental
     */
    toCreateKnowledgeBase() {
        return this.to('CreateKnowledgeBase');
    }
    /**
     * (experimental) Grants permission to create a session.
     *
     * Access Level: Write
     *
     * Possible conditions:
     * - .ifAwsTagKeys()
     * - .ifAwsRequestTag()
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_CreateSession.html
     *
     * @experimental
     */
    toCreateSession() {
        return this.to('CreateSession');
    }
    /**
     * (experimental) Grants permission to delete an assistant.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_DeleteAssistant.html
     *
     * @experimental
     */
    toDeleteAssistant() {
        return this.to('DeleteAssistant');
    }
    /**
     * (experimental) Grants permission to delete an assistant association.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_DeleteAssistantAssociation.html
     *
     * @experimental
     */
    toDeleteAssistantAssociation() {
        return this.to('DeleteAssistantAssociation');
    }
    /**
     * (experimental) Grants permission to delete content.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_DeleteContent.html
     *
     * @experimental
     */
    toDeleteContent() {
        return this.to('DeleteContent');
    }
    /**
     * (experimental) Grants permission to delete a knowledge base.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_DeleteKnowledgeBase.html
     *
     * @experimental
     */
    toDeleteKnowledgeBase() {
        return this.to('DeleteKnowledgeBase');
    }
    /**
     * (experimental) Grants permission to retrieve information about an assistant.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_GetAssistant.html
     *
     * @experimental
     */
    toGetAssistant() {
        return this.to('GetAssistant');
    }
    /**
     * (experimental) Grants permission to retrieve information about an assistant association.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_GetAssistantAssociation.html
     *
     * @experimental
     */
    toGetAssistantAssociation() {
        return this.to('GetAssistantAssociation');
    }
    /**
     * (experimental) Grants permission to retrieve content, including a pre-signed URL to download the content.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_GetContent.html
     *
     * @experimental
     */
    toGetContent() {
        return this.to('GetContent');
    }
    /**
     * (experimental) Grants permission to retrieve summary information about the content.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_GetContentSummary.html
     *
     * @experimental
     */
    toGetContentSummary() {
        return this.to('GetContentSummary');
    }
    /**
     * (experimental) Grants permission to retrieve information about the knowledge base.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_GetKnowledgeBase.html
     *
     * @experimental
     */
    toGetKnowledgeBase() {
        return this.to('GetKnowledgeBase');
    }
    /**
     * (experimental) Grants permission to retrieve recommendations for the specified session.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_GetRecommendations.html
     *
     * @experimental
     */
    toGetRecommendations() {
        return this.to('GetRecommendations');
    }
    /**
     * (experimental) Grants permission to retrieve information for a specified session.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_GetSession.html
     *
     * @experimental
     */
    toGetSession() {
        return this.to('GetSession');
    }
    /**
     * (experimental) Grants permission to list information about assistant associations.
     *
     * Access Level: List
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_ListAssistantAssociations.html
     *
     * @experimental
     */
    toListAssistantAssociations() {
        return this.to('ListAssistantAssociations');
    }
    /**
     * (experimental) Grants permission to list information about assistants.
     *
     * Access Level: List
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_ListAssistants.html
     *
     * @experimental
     */
    toListAssistants() {
        return this.to('ListAssistants');
    }
    /**
     * (experimental) Grants permission to list the content with a knowledge base.
     *
     * Access Level: List
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_ListContents.html
     *
     * @experimental
     */
    toListContents() {
        return this.to('ListContents');
    }
    /**
     * (experimental) Grants permission to list information about knowledge bases.
     *
     * Access Level: List
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_ListKnowledgeBases.html
     *
     * @experimental
     */
    toListKnowledgeBases() {
        return this.to('ListKnowledgeBases');
    }
    /**
     * (experimental) Grants permission to list the tags for the specified resource.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_ListTagsForResource.html
     *
     * @experimental
     */
    toListTagsForResource() {
        return this.to('ListTagsForResource');
    }
    /**
     * (experimental) Grants permission to remove the specified recommendations from the specified assistant's queue of newly available recommendations.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_NotifyRecommendationsReceived.html
     *
     * @experimental
     */
    toNotifyRecommendationsReceived() {
        return this.to('NotifyRecommendationsReceived');
    }
    /**
     * (experimental) Grants permission to perform a manual search against the specified assistant.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_QueryAssistant.html
     *
     * @experimental
     */
    toQueryAssistant() {
        return this.to('QueryAssistant');
    }
    /**
     * (experimental) Grants permission to remove a URI template from a knowledge base.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_RemoveKnowledgeBaseTemplateUri.html
     *
     * @experimental
     */
    toRemoveKnowledgeBaseTemplateUri() {
        return this.to('RemoveKnowledgeBaseTemplateUri');
    }
    /**
     * (experimental) Grants permission to search for content referencing a specified knowledge base.
     *
     * Can be used to get a specific content resource by its name
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_SearchContent.html
     *
     * @experimental
     */
    toSearchContent() {
        return this.to('SearchContent');
    }
    /**
     * (experimental) Grants permission to search for sessions referencing a specified assistant.
     *
     * Can be used to et a specific session resource by its name
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_SearchSessions.html
     *
     * @experimental
     */
    toSearchSessions() {
        return this.to('SearchSessions');
    }
    /**
     * (experimental) Grants permission to get a URL to upload content to a knowledge base.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_StartContentUpload.html
     *
     * @experimental
     */
    toStartContentUpload() {
        return this.to('StartContentUpload');
    }
    /**
     * (experimental) Grants permission to add the specified tags to the specified resource.
     *
     * Access Level: Tagging
     *
     * Possible conditions:
     * - .ifAwsTagKeys()
     * - .ifAwsRequestTag()
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_TagResource.html
     *
     * @experimental
     */
    toTagResource() {
        return this.to('TagResource');
    }
    /**
     * (experimental) Grants permission to remove the specified tags from the specified resource.
     *
     * Access Level: Tagging
     *
     * Possible conditions:
     * - .ifAwsTagKeys()
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_UntagResource.html
     *
     * @experimental
     */
    toUntagResource() {
        return this.to('UntagResource');
    }
    /**
     * (experimental) Grants permission to update information about the content.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_UpdateContent.html
     *
     * @experimental
     */
    toUpdateContent() {
        return this.to('UpdateContent');
    }
    /**
     * (experimental) Grants permission to update the template URI of a knowledge base.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_UpdateKnowledgeBaseTemplateUri.html
     *
     * @experimental
     */
    toUpdateKnowledgeBaseTemplateUri() {
        return this.to('UpdateKnowledgeBaseTemplateUri');
    }
    /**
     * (experimental) Adds a resource of type Assistant to the statement.
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_AssistantData.html
     *
     * @param assistantId - Identifier for the assistantId.
     * @param account - Account of the resource;
     * @param region - Region of the resource;
     * @param partition - Partition of the AWS account [aws, aws-cn, aws-us-gov]; defaults to `aws`.
     * @experimental
     */
    onAssistant(assistantId, account, region, partition) {
        return this.on(`arn:${partition || 'aws'}:wisdom:${region || '*'}:${account || '*'}:assistant/${assistantId}`);
    }
    /**
     * (experimental) Adds a resource of type AssistantAssociation to the statement.
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_AssistantAssociationData.html
     *
     * @param assistantId - Identifier for the assistantId.
     * @param assistantAssociationId - Identifier for the assistantAssociationId.
     * @param account - Account of the resource;
     * @param region - Region of the resource;
     * @param partition - Partition of the AWS account [aws, aws-cn, aws-us-gov]; defaults to `aws`.
     * @experimental
     */
    onAssistantAssociation(assistantId, assistantAssociationId, account, region, partition) {
        return this.on(`arn:${partition || 'aws'}:wisdom:${region || '*'}:${account || '*'}:association/${assistantId}/${assistantAssociationId}`);
    }
    /**
     * (experimental) Adds a resource of type Content to the statement.
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_ContentData.html
     *
     * @param knowledgeBaseId - Identifier for the knowledgeBaseId.
     * @param contentId - Identifier for the contentId.
     * @param account - Account of the resource;
     * @param region - Region of the resource;
     * @param partition - Partition of the AWS account [aws, aws-cn, aws-us-gov]; defaults to `aws`.
     * @experimental
     */
    onContent(knowledgeBaseId, contentId, account, region, partition) {
        return this.on(`arn:${partition || 'aws'}:wisdom:${region || '*'}:${account || '*'}:content/${knowledgeBaseId}/${contentId}`);
    }
    /**
     * (experimental) Adds a resource of type KnowledgeBase to the statement.
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_KnowledgeBaseData.html
     *
     * @param knowledgeBaseId - Identifier for the knowledgeBaseId.
     * @param account - Account of the resource;
     * @param region - Region of the resource;
     * @param partition - Partition of the AWS account [aws, aws-cn, aws-us-gov]; defaults to `aws`.
     * @experimental
     */
    onKnowledgeBase(knowledgeBaseId, account, region, partition) {
        return this.on(`arn:${partition || 'aws'}:wisdom:${region || '*'}:${account || '*'}:knowledge-base/${knowledgeBaseId}`);
    }
    /**
     * (experimental) Adds a resource of type Session to the statement.
     *
     * https://docs.aws.amazon.com/wisdom/latest/APIReference/API_SessionData.html
     *
     * @param assistantId - Identifier for the assistantId.
     * @param sessionId - Identifier for the sessionId.
     * @param account - Account of the resource;
     * @param region - Region of the resource;
     * @param partition - Partition of the AWS account [aws, aws-cn, aws-us-gov]; defaults to `aws`.
     * @experimental
     */
    onSession(assistantId, sessionId, account, region, partition) {
        return this.on(`arn:${partition || 'aws'}:wisdom:${region || '*'}:${account || '*'}:session/${assistantId}/${sessionId}`);
    }
}
exports.Wisdom = Wisdom;
_a = JSII_RTTI_SYMBOL_1;
Wisdom[_a] = { fqn: "cdk-iam-floyd.Wisdom", version: "0.300.0" };
