// cdk-iam-floyd/lib/generated/groundtruthlabeling.js
"use strict";
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
exports.Groundtruthlabeling = void 0;
const JSII_RTTI_SYMBOL_1 = Symbol.for("jsii.rtti");
const shared_1 = require("../shared");
/**
 * (experimental) Statement provider for service [groundtruthlabeling](https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazongroundtruthlabeling.html).
 *
 * @experimental
 */
class Groundtruthlabeling extends shared_1.PolicyStatement {
    /**
     * (experimental) Statement provider for service [groundtruthlabeling](https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazongroundtruthlabeling.html).
     *
     * @experimental
     */
    constructor(props) {
        super(props);
        /**
         * (experimental) Holds the prefix of the service actions, e.g. `ec2`.
         *
         * @experimental
         */
        this.servicePrefix = 'groundtruthlabeling';
        /**
         * @experimental
         */
        this.accessLevelList = {
            Write: [
                'AssociatePatchToManifestJob',
                'RunFilterOrSampleDatasetJob',
                'RunGenerateManifestByCrawlingJob'
            ],
            Read: [
                'DescribeConsoleJob',
                'ListDatasetObjects'
            ]
        };
    }
    /**
     * (experimental) Grants permission to associate a patch file with the manifest file to update the manifest file.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/sagemaker/latest/dg/sms-data-input.html#sms-console-create-manifest-file
     *
     * @experimental
     */
    toAssociatePatchToManifestJob() {
        return this.to('AssociatePatchToManifestJob');
    }
    /**
     * (experimental) Grants permission to get status of GroundTruthLabeling Jobs.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/sagemaker/latest/dg/sms-data-input.html#sms-console-create-manifest-file
     *
     * @experimental
     */
    toDescribeConsoleJob() {
        return this.to('DescribeConsoleJob');
    }
    /**
     * (experimental) Grants permission to list dataset objects in a manifest file.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/sagemaker/latest/dg/sms-data-input.html#sms-console-create-manifest-file
     *
     * @experimental
     */
    toListDatasetObjects() {
        return this.to('ListDatasetObjects');
    }
    /**
     * (experimental) Grants permission to filter records from a manifest file using S3 select. Get sample entries based on random sampling.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/sagemaker/latest/dg/sms-data-input.html#sms-data-filtering
     *
     * @experimental
     */
    toRunFilterOrSampleDatasetJob() {
        return this.to('RunFilterOrSampleDatasetJob');
    }
    /**
     * (experimental) Grants permission to list a S3 prefix and create manifest files from objects in that location.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/sagemaker/latest/dg/sms-data-input.html#sms-console-create-manifest-file
     *
     * @experimental
     */
    toRunGenerateManifestByCrawlingJob() {
        return this.to('RunGenerateManifestByCrawlingJob');
    }
}
exports.Groundtruthlabeling = Groundtruthlabeling;
_a = JSII_RTTI_SYMBOL_1;
Groundtruthlabeling[_a] = { fqn: "cdk-iam-floyd.Groundtruthlabeling", version: "0.300.0" };
