// cdk-iam-floyd/lib/generated/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var alexaforbusiness_1 = require("./alexaforbusiness");
Object.defineProperty(exports, "A4b", { enumerable: true, get: function () { return alexaforbusiness_1.A4b; } });
var apigateway_1 = require("./apigateway");
Object.defineProperty(exports, "ExecuteApi", { enumerable: true, get: function () { return apigateway_1.ExecuteApi; } });
var apigatewaymanagement_1 = require("./apigatewaymanagement");
Object.defineProperty(exports, "Apigateway", { enumerable: true, get: function () { return apigatewaymanagement_1.Apigateway; } });
var apigatewaymanagementv2_1 = require("./apigatewaymanagementv2");
Object.defineProperty(exports, "ApigatewayV2", { enumerable: true, get: function () { return apigatewaymanagementv2_1.ApigatewayV2; } });
var appflow_1 = require("./appflow");
Object.defineProperty(exports, "Appflow", { enumerable: true, get: function () { return appflow_1.Appflow; } });
var appintegrations_1 = require("./appintegrations");
Object.defineProperty(exports, "AppIntegrations", { enumerable: true, get: function () { return appintegrations_1.AppIntegrations; } });
var appstream2_0_1 = require("./appstream2-0");
Object.defineProperty(exports, "Appstream", { enumerable: true, get: function () { return appstream2_0_1.Appstream; } });
var athena_1 = require("./athena");
Object.defineProperty(exports, "Athena", { enumerable: true, get: function () { return athena_1.Athena; } });
var braket_1 = require("./braket");
Object.defineProperty(exports, "Braket", { enumerable: true, get: function () { return braket_1.Braket; } });
var chime_1 = require("./chime");
Object.defineProperty(exports, "Chime", { enumerable: true, get: function () { return chime_1.Chime; } });
var clouddirectory_1 = require("./clouddirectory");
Object.defineProperty(exports, "Clouddirectory", { enumerable: true, get: function () { return clouddirectory_1.Clouddirectory; } });
var cloudfront_1 = require("./cloudfront");
Object.defineProperty(exports, "Cloudfront", { enumerable: true, get: function () { return cloudfront_1.Cloudfront; } });
var cloudsearch_1 = require("./cloudsearch");
Object.defineProperty(exports, "Cloudsearch", { enumerable: true, get: function () { return cloudsearch_1.Cloudsearch; } });
var cloudwatch_1 = require("./cloudwatch");
Object.defineProperty(exports, "Cloudwatch", { enumerable: true, get: function () { return cloudwatch_1.Cloudwatch; } });
var cloudwatchlogs_1 = require("./cloudwatchlogs");
Object.defineProperty(exports, "Logs", { enumerable: true, get: function () { return cloudwatchlogs_1.Logs; } });
var cloudwatchsynthetics_1 = require("./cloudwatchsynthetics");
Object.defineProperty(exports, "Synthetics", { enumerable: true, get: function () { return cloudwatchsynthetics_1.Synthetics; } });
var codeguru_1 = require("./codeguru");
Object.defineProperty(exports, "Codeguru", { enumerable: true, get: function () { return codeguru_1.Codeguru; } });
var codeguruprofiler_1 = require("./codeguruprofiler");
Object.defineProperty(exports, "CodeguruProfiler", { enumerable: true, get: function () { return codeguruprofiler_1.CodeguruProfiler; } });
var codegurureviewer_1 = require("./codegurureviewer");
Object.defineProperty(exports, "CodeguruReviewer", { enumerable: true, get: function () { return codegurureviewer_1.CodeguruReviewer; } });
var cognitoidentity_1 = require("./cognitoidentity");
Object.defineProperty(exports, "CognitoIdentity", { enumerable: true, get: function () { return cognitoidentity_1.CognitoIdentity; } });
var cognitosync_1 = require("./cognitosync");
Object.defineProperty(exports, "CognitoSync", { enumerable: true, get: function () { return cognitosync_1.CognitoSync; } });
var cognitouserpools_1 = require("./cognitouserpools");
Object.defineProperty(exports, "CognitoIdp", { enumerable: true, get: function () { return cognitouserpools_1.CognitoIdp; } });
var comprehend_1 = require("./comprehend");
Object.defineProperty(exports, "Comprehend", { enumerable: true, get: function () { return comprehend_1.Comprehend; } });
var comprehendmedical_1 = require("./comprehendmedical");
Object.defineProperty(exports, "Comprehendmedical", { enumerable: true, get: function () { return comprehendmedical_1.Comprehendmedical; } });
var connect_1 = require("./connect");
Object.defineProperty(exports, "Connect", { enumerable: true, get: function () { return connect_1.Connect; } });
var connectcustomerprofiles_1 = require("./connectcustomerprofiles");
Object.defineProperty(exports, "Profile", { enumerable: true, get: function () { return connectcustomerprofiles_1.Profile; } });
var connectvoiceid_1 = require("./connectvoiceid");
Object.defineProperty(exports, "Voiceid", { enumerable: true, get: function () { return connectvoiceid_1.Voiceid; } });
var connectwisdom_1 = require("./connectwisdom");
Object.defineProperty(exports, "Wisdom", { enumerable: true, get: function () { return connectwisdom_1.Wisdom; } });
var datalifecyclemanager_1 = require("./datalifecyclemanager");
Object.defineProperty(exports, "Dlm", { enumerable: true, get: function () { return datalifecyclemanager_1.Dlm; } });
var detective_1 = require("./detective");
Object.defineProperty(exports, "Detective", { enumerable: true, get: function () { return detective_1.Detective; } });
var devopsguru_1 = require("./devopsguru");
Object.defineProperty(exports, "DevopsGuru", { enumerable: true, get: function () { return devopsguru_1.DevopsGuru; } });
var dynamodb_1 = require("./dynamodb");
Object.defineProperty(exports, "Dynamodb", { enumerable: true, get: function () { return dynamodb_1.Dynamodb; } });
var dynamodbacceleratordax_1 = require("./dynamodbacceleratordax");
Object.defineProperty(exports, "Dax", { enumerable: true, get: function () { return dynamodbacceleratordax_1.Dax; } });
var ec2_1 = require("./ec2");
Object.defineProperty(exports, "Ec2", { enumerable: true, get: function () { return ec2_1.Ec2; } });
var ec2autoscaling_1 = require("./ec2autoscaling");
Object.defineProperty(exports, "Autoscaling", { enumerable: true, get: function () { return ec2autoscaling_1.Autoscaling; } });
var ec2imagebuilder_1 = require("./ec2imagebuilder");
Object.defineProperty(exports, "Imagebuilder", { enumerable: true, get: function () { return ec2imagebuilder_1.Imagebuilder; } });
var ec2instanceconnect_1 = require("./ec2instanceconnect");
Object.defineProperty(exports, "Ec2InstanceConnect", { enumerable: true, get: function () { return ec2instanceconnect_1.Ec2InstanceConnect; } });
var elasticache_1 = require("./elasticache");
Object.defineProperty(exports, "Elasticache", { enumerable: true, get: function () { return elasticache_1.Elasticache; } });
var elasticblockstore_1 = require("./elasticblockstore");
Object.defineProperty(exports, "Ebs", { enumerable: true, get: function () { return elasticblockstore_1.Ebs; } });
var elasticcontainerregistry_1 = require("./elasticcontainerregistry");
Object.defineProperty(exports, "Ecr", { enumerable: true, get: function () { return elasticcontainerregistry_1.Ecr; } });
var elasticcontainerregistrypublic_1 = require("./elasticcontainerregistrypublic");
Object.defineProperty(exports, "EcrPublic", { enumerable: true, get: function () { return elasticcontainerregistrypublic_1.EcrPublic; } });
var elasticcontainerservice_1 = require("./elasticcontainerservice");
Object.defineProperty(exports, "Ecs", { enumerable: true, get: function () { return elasticcontainerservice_1.Ecs; } });
var elasticfilesystem_1 = require("./elasticfilesystem");
Object.defineProperty(exports, "Elasticfilesystem", { enumerable: true, get: function () { return elasticfilesystem_1.Elasticfilesystem; } });
var elasticinference_1 = require("./elasticinference");
Object.defineProperty(exports, "ElasticInference", { enumerable: true, get: function () { return elasticinference_1.ElasticInference; } });
var elastickubernetesservice_1 = require("./elastickubernetesservice");
Object.defineProperty(exports, "Eks", { enumerable: true, get: function () { return elastickubernetesservice_1.Eks; } });
var elasticmapreduce_1 = require("./elasticmapreduce");
Object.defineProperty(exports, "Elasticmapreduce", { enumerable: true, get: function () { return elasticmapreduce_1.Elasticmapreduce; } });
var elastictranscoder_1 = require("./elastictranscoder");
Object.defineProperty(exports, "Elastictranscoder", { enumerable: true, get: function () { return elastictranscoder_1.Elastictranscoder; } });
var emroneksemrcontainers_1 = require("./emroneksemrcontainers");
Object.defineProperty(exports, "EmrContainers", { enumerable: true, get: function () { return emroneksemrcontainers_1.EmrContainers; } });
var eventbridge_1 = require("./eventbridge");
Object.defineProperty(exports, "Events", { enumerable: true, get: function () { return eventbridge_1.Events; } });
var eventbridgeschemas_1 = require("./eventbridgeschemas");
Object.defineProperty(exports, "Schemas", { enumerable: true, get: function () { return eventbridgeschemas_1.Schemas; } });
var finspace_1 = require("./finspace");
Object.defineProperty(exports, "Finspace", { enumerable: true, get: function () { return finspace_1.Finspace; } });
var forecast_1 = require("./forecast");
Object.defineProperty(exports, "Forecast", { enumerable: true, get: function () { return forecast_1.Forecast; } });
var frauddetector_1 = require("./frauddetector");
Object.defineProperty(exports, "Frauddetector", { enumerable: true, get: function () { return frauddetector_1.Frauddetector; } });
var freertos_1 = require("./freertos");
Object.defineProperty(exports, "Freertos", { enumerable: true, get: function () { return freertos_1.Freertos; } });
var fsx_1 = require("./fsx");
Object.defineProperty(exports, "Fsx", { enumerable: true, get: function () { return fsx_1.Fsx; } });
var gamelift_1 = require("./gamelift");
Object.defineProperty(exports, "Gamelift", { enumerable: true, get: function () { return gamelift_1.Gamelift; } });
var glacier_1 = require("./glacier");
Object.defineProperty(exports, "Glacier", { enumerable: true, get: function () { return glacier_1.Glacier; } });
var groundtruthlabeling_1 = require("./groundtruthlabeling");
Object.defineProperty(exports, "Groundtruthlabeling", { enumerable: true, get: function () { return groundtruthlabeling_1.Groundtruthlabeling; } });
var guardduty_1 = require("./guardduty");
Object.defineProperty(exports, "Guardduty", { enumerable: true, get: function () { return guardduty_1.Guardduty; } });
var healthlake_1 = require("./healthlake");
Object.defineProperty(exports, "Healthlake", { enumerable: true, get: function () { return healthlake_1.Healthlake; } });
var honeycode_1 = require("./honeycode");
Object.defineProperty(exports, "Honeycode", { enumerable: true, get: function () { return honeycode_1.Honeycode; } });
var inspector_1 = require("./inspector");
Object.defineProperty(exports, "Inspector", { enumerable: true, get: function () { return inspector_1.Inspector; } });
var inspector2_1 = require("./inspector2");
Object.defineProperty(exports, "Inspector2", { enumerable: true, get: function () { return inspector2_1.Inspector2; } });
var interactivevideoservice_1 = require("./interactivevideoservice");
Object.defineProperty(exports, "Ivs", { enumerable: true, get: function () { return interactivevideoservice_1.Ivs; } });
var kendra_1 = require("./kendra");
Object.defineProperty(exports, "Kendra", { enumerable: true, get: function () { return kendra_1.Kendra; } });
var keyspacesforapachecassandra_1 = require("./keyspacesforapachecassandra");
Object.defineProperty(exports, "Cassandra", { enumerable: true, get: function () { return keyspacesforapachecassandra_1.Cassandra; } });
var kinesis_1 = require("./kinesis");
Object.defineProperty(exports, "Kinesis", { enumerable: true, get: function () { return kinesis_1.Kinesis; } });
var kinesisanalytics_1 = require("./kinesisanalytics");
Object.defineProperty(exports, "Kinesisanalytics", { enumerable: true, get: function () { return kinesisanalytics_1.Kinesisanalytics; } });
var kinesisanalyticsv2_1 = require("./kinesisanalyticsv2");
Object.defineProperty(exports, "KinesisanalyticsV2", { enumerable: true, get: function () { return kinesisanalyticsv2_1.KinesisanalyticsV2; } });
var kinesisfirehose_1 = require("./kinesisfirehose");
Object.defineProperty(exports, "Firehose", { enumerable: true, get: function () { return kinesisfirehose_1.Firehose; } });
var kinesisvideostreams_1 = require("./kinesisvideostreams");
Object.defineProperty(exports, "Kinesisvideo", { enumerable: true, get: function () { return kinesisvideostreams_1.Kinesisvideo; } });
var lex_1 = require("./lex");
Object.defineProperty(exports, "Lex", { enumerable: true, get: function () { return lex_1.Lex; } });
var lexv2_1 = require("./lexv2");
Object.defineProperty(exports, "LexV2", { enumerable: true, get: function () { return lexv2_1.LexV2; } });
var lightsail_1 = require("./lightsail");
Object.defineProperty(exports, "Lightsail", { enumerable: true, get: function () { return lightsail_1.Lightsail; } });
var location_1 = require("./location");
Object.defineProperty(exports, "Geo", { enumerable: true, get: function () { return location_1.Geo; } });
var lookoutforequipment_1 = require("./lookoutforequipment");
Object.defineProperty(exports, "Lookoutequipment", { enumerable: true, get: function () { return lookoutforequipment_1.Lookoutequipment; } });
var lookoutformetrics_1 = require("./lookoutformetrics");
Object.defineProperty(exports, "Lookoutmetrics", { enumerable: true, get: function () { return lookoutformetrics_1.Lookoutmetrics; } });
var lookoutforvision_1 = require("./lookoutforvision");
Object.defineProperty(exports, "Lookoutvision", { enumerable: true, get: function () { return lookoutforvision_1.Lookoutvision; } });
var machinelearning_1 = require("./machinelearning");
Object.defineProperty(exports, "Machinelearning", { enumerable: true, get: function () { return machinelearning_1.Machinelearning; } });
var macie_1 = require("./macie");
Object.defineProperty(exports, "Macie2", { enumerable: true, get: function () { return macie_1.Macie2; } });
var macieclassic_1 = require("./macieclassic");
Object.defineProperty(exports, "Macie", { enumerable: true, get: function () { return macieclassic_1.Macie; } });
var managedblockchain_1 = require("./managedblockchain");
Object.defineProperty(exports, "Managedblockchain", { enumerable: true, get: function () { return managedblockchain_1.Managedblockchain; } });
var managedgrafana_1 = require("./managedgrafana");
Object.defineProperty(exports, "Grafana", { enumerable: true, get: function () { return managedgrafana_1.Grafana; } });
var managedserviceforprometheus_1 = require("./managedserviceforprometheus");
Object.defineProperty(exports, "Aps", { enumerable: true, get: function () { return managedserviceforprometheus_1.Aps; } });
var managedstreamingforapachekafka_1 = require("./managedstreamingforapachekafka");
Object.defineProperty(exports, "Kafka", { enumerable: true, get: function () { return managedstreamingforapachekafka_1.Kafka; } });
var managedstreamingforkafkaconnect_1 = require("./managedstreamingforkafkaconnect");
Object.defineProperty(exports, "Kafkaconnect", { enumerable: true, get: function () { return managedstreamingforkafkaconnect_1.Kafkaconnect; } });
var managedworkflowsforapacheairflow_1 = require("./managedworkflowsforapacheairflow");
Object.defineProperty(exports, "Airflow", { enumerable: true, get: function () { return managedworkflowsforapacheairflow_1.Airflow; } });
var mechanicalturk_1 = require("./mechanicalturk");
Object.defineProperty(exports, "Mechanicalturk", { enumerable: true, get: function () { return mechanicalturk_1.Mechanicalturk; } });
var mediaimport_1 = require("./mediaimport");
Object.defineProperty(exports, "Mediaimport", { enumerable: true, get: function () { return mediaimport_1.Mediaimport; } });
var memorydb_1 = require("./memorydb");
Object.defineProperty(exports, "Memorydb", { enumerable: true, get: function () { return memorydb_1.Memorydb; } });
var messagedeliveryservice_1 = require("./messagedeliveryservice");
Object.defineProperty(exports, "Ec2messages", { enumerable: true, get: function () { return messagedeliveryservice_1.Ec2messages; } });
var mobileanalytics_1 = require("./mobileanalytics");
Object.defineProperty(exports, "Mobileanalytics", { enumerable: true, get: function () { return mobileanalytics_1.Mobileanalytics; } });
var monitron_1 = require("./monitron");
Object.defineProperty(exports, "Monitron", { enumerable: true, get: function () { return monitron_1.Monitron; } });
var mq_1 = require("./mq");
Object.defineProperty(exports, "Mq", { enumerable: true, get: function () { return mq_1.Mq; } });
var neptune_1 = require("./neptune");
Object.defineProperty(exports, "NeptuneDb", { enumerable: true, get: function () { return neptune_1.NeptuneDb; } });
var nimblestudio_1 = require("./nimblestudio");
Object.defineProperty(exports, "Nimble", { enumerable: true, get: function () { return nimblestudio_1.Nimble; } });
var opensearchservicesuccessortoamazonelasticsearchservice_1 = require("./opensearchservicesuccessortoamazonelasticsearchservice");
Object.defineProperty(exports, "Es", { enumerable: true, get: function () { return opensearchservicesuccessortoamazonelasticsearchservice_1.Es; } });
var personalize_1 = require("./personalize");
Object.defineProperty(exports, "Personalize", { enumerable: true, get: function () { return personalize_1.Personalize; } });
var pinpoint_1 = require("./pinpoint");
Object.defineProperty(exports, "Mobiletargeting", { enumerable: true, get: function () { return pinpoint_1.Mobiletargeting; } });
var pinpointemailservice_1 = require("./pinpointemailservice");
Object.defineProperty(exports, "SesPinpoint", { enumerable: true, get: function () { return pinpointemailservice_1.SesPinpoint; } });
var pinpointsmsandvoiceservice_1 = require("./pinpointsmsandvoiceservice");
Object.defineProperty(exports, "SmsVoice", { enumerable: true, get: function () { return pinpointsmsandvoiceservice_1.SmsVoice; } });
var polly_1 = require("./polly");
Object.defineProperty(exports, "Polly", { enumerable: true, get: function () { return polly_1.Polly; } });
var qldb_1 = require("./qldb");
Object.defineProperty(exports, "Qldb", { enumerable: true, get: function () { return qldb_1.Qldb; } });
var quicksight_1 = require("./quicksight");
Object.defineProperty(exports, "Quicksight", { enumerable: true, get: function () { return quicksight_1.Quicksight; } });
var rds_1 = require("./rds");
Object.defineProperty(exports, "Rds", { enumerable: true, get: function () { return rds_1.Rds; } });
var rdsdataapi_1 = require("./rdsdataapi");
Object.defineProperty(exports, "RdsData", { enumerable: true, get: function () { return rdsdataapi_1.RdsData; } });
var rdsiamauthentication_1 = require("./rdsiamauthentication");
Object.defineProperty(exports, "RdsDb", { enumerable: true, get: function () { return rdsiamauthentication_1.RdsDb; } });
var redshift_1 = require("./redshift");
Object.defineProperty(exports, "Redshift", { enumerable: true, get: function () { return redshift_1.Redshift; } });
var redshiftdataapi_1 = require("./redshiftdataapi");
Object.defineProperty(exports, "RedshiftData", { enumerable: true, get: function () { return redshiftdataapi_1.RedshiftData; } });
var rekognition_1 = require("./rekognition");
Object.defineProperty(exports, "Rekognition", { enumerable: true, get: function () { return rekognition_1.Rekognition; } });
var resourcegrouptaggingapi_1 = require("./resourcegrouptaggingapi");
Object.defineProperty(exports, "Tag", { enumerable: true, get: function () { return resourcegrouptaggingapi_1.Tag; } });
var route53_1 = require("./route53");
Object.defineProperty(exports, "Route53", { enumerable: true, get: function () { return route53_1.Route53; } });
var route53domains_1 = require("./route53domains");
Object.defineProperty(exports, "Route53domains", { enumerable: true, get: function () { return route53domains_1.Route53domains; } });
var route53recoverycluster_1 = require("./route53recoverycluster");
Object.defineProperty(exports, "Route53RecoveryCluster", { enumerable: true, get: function () { return route53recoverycluster_1.Route53RecoveryCluster; } });
var route53recoverycontrols_1 = require("./route53recoverycontrols");
Object.defineProperty(exports, "Route53RecoveryControlConfig", { enumerable: true, get: function () { return route53recoverycontrols_1.Route53RecoveryControlConfig; } });
var route53recoveryreadiness_1 = require("./route53recoveryreadiness");
Object.defineProperty(exports, "Route53RecoveryReadiness", { enumerable: true, get: function () { return route53recoveryreadiness_1.Route53RecoveryReadiness; } });
var route53resolver_1 = require("./route53resolver");
Object.defineProperty(exports, "Route53resolver", { enumerable: true, get: function () { return route53resolver_1.Route53resolver; } });
var s3_1 = require("./s3");
Object.defineProperty(exports, "S3", { enumerable: true, get: function () { return s3_1.S3; } });
var s3objectlambda_1 = require("./s3objectlambda");
Object.defineProperty(exports, "S3ObjectLambda", { enumerable: true, get: function () { return s3objectlambda_1.S3ObjectLambda; } });
var s3onoutposts_1 = require("./s3onoutposts");
Object.defineProperty(exports, "S3Outposts", { enumerable: true, get: function () { return s3onoutposts_1.S3Outposts; } });
var sagemaker_1 = require("./sagemaker");
Object.defineProperty(exports, "Sagemaker", { enumerable: true, get: function () { return sagemaker_1.Sagemaker; } });
var ses_1 = require("./ses");
Object.defineProperty(exports, "Ses", { enumerable: true, get: function () { return ses_1.Ses; } });
var sessionmanagermessagegatewayservice_1 = require("./sessionmanagermessagegatewayservice");
Object.defineProperty(exports, "Ssmmessages", { enumerable: true, get: function () { return sessionmanagermessagegatewayservice_1.Ssmmessages; } });
var simpledb_1 = require("./simpledb");
Object.defineProperty(exports, "Sdb", { enumerable: true, get: function () { return simpledb_1.Sdb; } });
var simpleemailservicev2_1 = require("./simpleemailservicev2");
Object.defineProperty(exports, "SesV2", { enumerable: true, get: function () { return simpleemailservicev2_1.SesV2; } });
var simpleworkflowservice_1 = require("./simpleworkflowservice");
Object.defineProperty(exports, "Swf", { enumerable: true, get: function () { return simpleworkflowservice_1.Swf; } });
var sns_1 = require("./sns");
Object.defineProperty(exports, "Sns", { enumerable: true, get: function () { return sns_1.Sns; } });
var sqs_1 = require("./sqs");
Object.defineProperty(exports, "Sqs", { enumerable: true, get: function () { return sqs_1.Sqs; } });
var storagegateway_1 = require("./storagegateway");
Object.defineProperty(exports, "Storagegateway", { enumerable: true, get: function () { return storagegateway_1.Storagegateway; } });
var sumerian_1 = require("./sumerian");
Object.defineProperty(exports, "Sumerian", { enumerable: true, get: function () { return sumerian_1.Sumerian; } });
var textract_1 = require("./textract");
Object.defineProperty(exports, "Textract", { enumerable: true, get: function () { return textract_1.Textract; } });
var timestream_1 = require("./timestream");
Object.defineProperty(exports, "Timestream", { enumerable: true, get: function () { return timestream_1.Timestream; } });
var transcribe_1 = require("./transcribe");
Object.defineProperty(exports, "Transcribe", { enumerable: true, get: function () { return transcribe_1.Transcribe; } });
var translate_1 = require("./translate");
Object.defineProperty(exports, "Translate", { enumerable: true, get: function () { return translate_1.Translate; } });
var workdocs_1 = require("./workdocs");
Object.defineProperty(exports, "Workdocs", { enumerable: true, get: function () { return workdocs_1.Workdocs; } });
var worklink_1 = require("./worklink");
Object.defineProperty(exports, "Worklink", { enumerable: true, get: function () { return worklink_1.Worklink; } });
var workmail_1 = require("./workmail");
Object.defineProperty(exports, "Workmail", { enumerable: true, get: function () { return workmail_1.Workmail; } });
var workmailmessageflow_1 = require("./workmailmessageflow");
Object.defineProperty(exports, "Workmailmessageflow", { enumerable: true, get: function () { return workmailmessageflow_1.Workmailmessageflow; } });
var workspaces_1 = require("./workspaces");
Object.defineProperty(exports, "Workspaces", { enumerable: true, get: function () { return workspaces_1.Workspaces; } });
var workspacesapplicationmanager_1 = require("./workspacesapplicationmanager");
Object.defineProperty(exports, "Wam", { enumerable: true, get: function () { return workspacesapplicationmanager_1.Wam; } });
var apachekafkaapisforamazonmskclusters_1 = require("./apachekafkaapisforamazonmskclusters");
Object.defineProperty(exports, "KafkaCluster", { enumerable: true, get: function () { return apachekafkaapisforamazonmskclusters_1.KafkaCluster; } });
var applicationautoscaling_1 = require("./applicationautoscaling");
Object.defineProperty(exports, "ApplicationAutoscaling", { enumerable: true, get: function () { return applicationautoscaling_1.ApplicationAutoscaling; } });
var applicationdiscovery_1 = require("./applicationdiscovery");
Object.defineProperty(exports, "Discovery", { enumerable: true, get: function () { return applicationdiscovery_1.Discovery; } });
var applicationdiscoveryarsenal_1 = require("./applicationdiscoveryarsenal");
Object.defineProperty(exports, "Arsenal", { enumerable: true, get: function () { return applicationdiscoveryarsenal_1.Arsenal; } });
var accountmanagement_1 = require("./accountmanagement");
Object.defineProperty(exports, "Account", { enumerable: true, get: function () { return accountmanagement_1.Account; } });
var activate_1 = require("./activate");
Object.defineProperty(exports, "Activate", { enumerable: true, get: function () { return activate_1.Activate; } });
var amplify_1 = require("./amplify");
Object.defineProperty(exports, "Amplify", { enumerable: true, get: function () { return amplify_1.Amplify; } });
var amplifyadmin_1 = require("./amplifyadmin");
Object.defineProperty(exports, "Amplifybackend", { enumerable: true, get: function () { return amplifyadmin_1.Amplifybackend; } });
var appconfig_1 = require("./appconfig");
Object.defineProperty(exports, "Appconfig", { enumerable: true, get: function () { return appconfig_1.Appconfig; } });
var applicationcostprofilerservice_1 = require("./applicationcostprofilerservice");
Object.defineProperty(exports, "ApplicationCostProfiler", { enumerable: true, get: function () { return applicationcostprofilerservice_1.ApplicationCostProfiler; } });
var applicationmigrationservice_1 = require("./applicationmigrationservice");
Object.defineProperty(exports, "Mgn", { enumerable: true, get: function () { return applicationmigrationservice_1.Mgn; } });
var appmesh_1 = require("./appmesh");
Object.defineProperty(exports, "Appmesh", { enumerable: true, get: function () { return appmesh_1.Appmesh; } });
var appmeshpreview_1 = require("./appmeshpreview");
Object.defineProperty(exports, "AppmeshPreview", { enumerable: true, get: function () { return appmeshpreview_1.AppmeshPreview; } });
var apprunner_1 = require("./apprunner");
Object.defineProperty(exports, "Apprunner", { enumerable: true, get: function () { return apprunner_1.Apprunner; } });
var appsync_1 = require("./appsync");
Object.defineProperty(exports, "Appsync", { enumerable: true, get: function () { return appsync_1.Appsync; } });
var artifact_1 = require("./artifact");
Object.defineProperty(exports, "Artifact", { enumerable: true, get: function () { return artifact_1.Artifact; } });
var auditmanager_1 = require("./auditmanager");
Object.defineProperty(exports, "Auditmanager", { enumerable: true, get: function () { return auditmanager_1.Auditmanager; } });
var autoscaling_1 = require("./autoscaling");
Object.defineProperty(exports, "AutoscalingPlans", { enumerable: true, get: function () { return autoscaling_1.AutoscalingPlans; } });
var backup_1 = require("./backup");
Object.defineProperty(exports, "Backup", { enumerable: true, get: function () { return backup_1.Backup; } });
var backupgateway_1 = require("./backupgateway");
Object.defineProperty(exports, "BackupGateway", { enumerable: true, get: function () { return backupgateway_1.BackupGateway; } });
var backupstorage_1 = require("./backupstorage");
Object.defineProperty(exports, "BackupStorage", { enumerable: true, get: function () { return backupstorage_1.BackupStorage; } });
var batch_1 = require("./batch");
Object.defineProperty(exports, "Batch", { enumerable: true, get: function () { return batch_1.Batch; } });
var billing_1 = require("./billing");
Object.defineProperty(exports, "AwsPortal", { enumerable: true, get: function () { return billing_1.AwsPortal; } });
var budgetservice_1 = require("./budgetservice");
Object.defineProperty(exports, "Budgets", { enumerable: true, get: function () { return budgetservice_1.Budgets; } });
var bugbust_1 = require("./bugbust");
Object.defineProperty(exports, "Bugbust", { enumerable: true, get: function () { return bugbust_1.Bugbust; } });
var certificatemanager_1 = require("./certificatemanager");
Object.defineProperty(exports, "Acm", { enumerable: true, get: function () { return certificatemanager_1.Acm; } });
var certificatemanagerprivatecertificateauthority_1 = require("./certificatemanagerprivatecertificateauthority");
Object.defineProperty(exports, "AcmPca", { enumerable: true, get: function () { return certificatemanagerprivatecertificateauthority_1.AcmPca; } });
var chatbot_1 = require("./chatbot");
Object.defineProperty(exports, "Chatbot", { enumerable: true, get: function () { return chatbot_1.Chatbot; } });
var cloud9_1 = require("./cloud9");
Object.defineProperty(exports, "Cloud9", { enumerable: true, get: function () { return cloud9_1.Cloud9; } });
var cloudcontrolapi_1 = require("./cloudcontrolapi");
Object.defineProperty(exports, "AwsCloudControlApi", { enumerable: true, get: function () { return cloudcontrolapi_1.AwsCloudControlApi; } });
var cloudformation_1 = require("./cloudformation");
Object.defineProperty(exports, "Cloudformation", { enumerable: true, get: function () { return cloudformation_1.Cloudformation; } });
var cloudhsm_1 = require("./cloudhsm");
Object.defineProperty(exports, "Cloudhsm", { enumerable: true, get: function () { return cloudhsm_1.Cloudhsm; } });
var cloudmap_1 = require("./cloudmap");
Object.defineProperty(exports, "Servicediscovery", { enumerable: true, get: function () { return cloudmap_1.Servicediscovery; } });
var cloudshell_1 = require("./cloudshell");
Object.defineProperty(exports, "Cloudshell", { enumerable: true, get: function () { return cloudshell_1.Cloudshell; } });
var cloudtrail_1 = require("./cloudtrail");
Object.defineProperty(exports, "Cloudtrail", { enumerable: true, get: function () { return cloudtrail_1.Cloudtrail; } });
var codeartifact_1 = require("./codeartifact");
Object.defineProperty(exports, "Codeartifact", { enumerable: true, get: function () { return codeartifact_1.Codeartifact; } });
var codebuild_1 = require("./codebuild");
Object.defineProperty(exports, "Codebuild", { enumerable: true, get: function () { return codebuild_1.Codebuild; } });
var codecommit_1 = require("./codecommit");
Object.defineProperty(exports, "Codecommit", { enumerable: true, get: function () { return codecommit_1.Codecommit; } });
var codedeploy_1 = require("./codedeploy");
Object.defineProperty(exports, "Codedeploy", { enumerable: true, get: function () { return codedeploy_1.Codedeploy; } });
var codepipeline_1 = require("./codepipeline");
Object.defineProperty(exports, "Codepipeline", { enumerable: true, get: function () { return codepipeline_1.Codepipeline; } });
var codestar_1 = require("./codestar");
Object.defineProperty(exports, "Codestar", { enumerable: true, get: function () { return codestar_1.Codestar; } });
var codestarconnections_1 = require("./codestarconnections");
Object.defineProperty(exports, "CodestarConnections", { enumerable: true, get: function () { return codestarconnections_1.CodestarConnections; } });
var codestarnotifications_1 = require("./codestarnotifications");
Object.defineProperty(exports, "CodestarNotifications", { enumerable: true, get: function () { return codestarnotifications_1.CodestarNotifications; } });
var computeoptimizer_1 = require("./computeoptimizer");
Object.defineProperty(exports, "ComputeOptimizer", { enumerable: true, get: function () { return computeoptimizer_1.ComputeOptimizer; } });
var config_1 = require("./config");
Object.defineProperty(exports, "Config", { enumerable: true, get: function () { return config_1.Config; } });
var connectorservice_1 = require("./connectorservice");
Object.defineProperty(exports, "Awsconnector", { enumerable: true, get: function () { return connectorservice_1.Awsconnector; } });
var controltower_1 = require("./controltower");
Object.defineProperty(exports, "Controltower", { enumerable: true, get: function () { return controltower_1.Controltower; } });
var costandusagereport_1 = require("./costandusagereport");
Object.defineProperty(exports, "Cur", { enumerable: true, get: function () { return costandusagereport_1.Cur; } });
var costexplorerservice_1 = require("./costexplorerservice");
Object.defineProperty(exports, "Ce", { enumerable: true, get: function () { return costexplorerservice_1.Ce; } });
var databasemigrationservice_1 = require("./databasemigrationservice");
Object.defineProperty(exports, "Dms", { enumerable: true, get: function () { return databasemigrationservice_1.Dms; } });
var dataexchange_1 = require("./dataexchange");
Object.defineProperty(exports, "Dataexchange", { enumerable: true, get: function () { return dataexchange_1.Dataexchange; } });
var datasync_1 = require("./datasync");
Object.defineProperty(exports, "Datasync", { enumerable: true, get: function () { return datasync_1.Datasync; } });
var deepcomposer_1 = require("./deepcomposer");
Object.defineProperty(exports, "Deepcomposer", { enumerable: true, get: function () { return deepcomposer_1.Deepcomposer; } });
var deeplens_1 = require("./deeplens");
Object.defineProperty(exports, "Deeplens", { enumerable: true, get: function () { return deeplens_1.Deeplens; } });
var deepracer_1 = require("./deepracer");
Object.defineProperty(exports, "Deepracer", { enumerable: true, get: function () { return deepracer_1.Deepracer; } });
var devicefarm_1 = require("./devicefarm");
Object.defineProperty(exports, "Devicefarm", { enumerable: true, get: function () { return devicefarm_1.Devicefarm; } });
var directconnect_1 = require("./directconnect");
Object.defineProperty(exports, "Directconnect", { enumerable: true, get: function () { return directconnect_1.Directconnect; } });
var directoryservice_1 = require("./directoryservice");
Object.defineProperty(exports, "Ds", { enumerable: true, get: function () { return directoryservice_1.Ds; } });
var elasticbeanstalk_1 = require("./elasticbeanstalk");
Object.defineProperty(exports, "Elasticbeanstalk", { enumerable: true, get: function () { return elasticbeanstalk_1.Elasticbeanstalk; } });
var elasticdisasterrecovery_1 = require("./elasticdisasterrecovery");
Object.defineProperty(exports, "Drs", { enumerable: true, get: function () { return elasticdisasterrecovery_1.Drs; } });
var elementalappliancesandsoftware_1 = require("./elementalappliancesandsoftware");
Object.defineProperty(exports, "ElementalAppliancesSoftware", { enumerable: true, get: function () { return elementalappliancesandsoftware_1.ElementalAppliancesSoftware; } });
var elementalappliancesandsoftwareactivationservice_1 = require("./elementalappliancesandsoftwareactivationservice");
Object.defineProperty(exports, "ElementalActivations", { enumerable: true, get: function () { return elementalappliancesandsoftwareactivationservice_1.ElementalActivations; } });
var elementalmediaconnect_1 = require("./elementalmediaconnect");
Object.defineProperty(exports, "Mediaconnect", { enumerable: true, get: function () { return elementalmediaconnect_1.Mediaconnect; } });
var elementalmediaconvert_1 = require("./elementalmediaconvert");
Object.defineProperty(exports, "Mediaconvert", { enumerable: true, get: function () { return elementalmediaconvert_1.Mediaconvert; } });
var elementalmedialive_1 = require("./elementalmedialive");
Object.defineProperty(exports, "Medialive", { enumerable: true, get: function () { return elementalmedialive_1.Medialive; } });
var elementalmediapackage_1 = require("./elementalmediapackage");
Object.defineProperty(exports, "Mediapackage", { enumerable: true, get: function () { return elementalmediapackage_1.Mediapackage; } });
var elementalmediapackagevod_1 = require("./elementalmediapackagevod");
Object.defineProperty(exports, "MediapackageVod", { enumerable: true, get: function () { return elementalmediapackagevod_1.MediapackageVod; } });
var elementalmediastore_1 = require("./elementalmediastore");
Object.defineProperty(exports, "Mediastore", { enumerable: true, get: function () { return elementalmediastore_1.Mediastore; } });
var elementalmediatailor_1 = require("./elementalmediatailor");
Object.defineProperty(exports, "Mediatailor", { enumerable: true, get: function () { return elementalmediatailor_1.Mediatailor; } });
var faultinjectionsimulator_1 = require("./faultinjectionsimulator");
Object.defineProperty(exports, "Fis", { enumerable: true, get: function () { return faultinjectionsimulator_1.Fis; } });
var firewallmanager_1 = require("./firewallmanager");
Object.defineProperty(exports, "Fms", { enumerable: true, get: function () { return firewallmanager_1.Fms; } });
var globalaccelerator_1 = require("./globalaccelerator");
Object.defineProperty(exports, "Globalaccelerator", { enumerable: true, get: function () { return globalaccelerator_1.Globalaccelerator; } });
var glue_1 = require("./glue");
Object.defineProperty(exports, "Glue", { enumerable: true, get: function () { return glue_1.Glue; } });
var gluedatabrew_1 = require("./gluedatabrew");
Object.defineProperty(exports, "Databrew", { enumerable: true, get: function () { return gluedatabrew_1.Databrew; } });
var groundstation_1 = require("./groundstation");
Object.defineProperty(exports, "Groundstation", { enumerable: true, get: function () { return groundstation_1.Groundstation; } });
var healthapisandnotifications_1 = require("./healthapisandnotifications");
Object.defineProperty(exports, "Health", { enumerable: true, get: function () { return healthapisandnotifications_1.Health; } });
var iamaccessanalyzer_1 = require("./iamaccessanalyzer");
Object.defineProperty(exports, "AccessAnalyzer", { enumerable: true, get: function () { return iamaccessanalyzer_1.AccessAnalyzer; } });
var identitystore_1 = require("./identitystore");
Object.defineProperty(exports, "Identitystore", { enumerable: true, get: function () { return identitystore_1.Identitystore; } });
var importexportdiskservice_1 = require("./importexportdiskservice");
Object.defineProperty(exports, "Importexport", { enumerable: true, get: function () { return importexportdiskservice_1.Importexport; } });
var iot_1 = require("./iot");
Object.defineProperty(exports, "Iot", { enumerable: true, get: function () { return iot_1.Iot; } });
var iot1_click_1 = require("./iot1-click");
Object.defineProperty(exports, "Iot1click", { enumerable: true, get: function () { return iot1_click_1.Iot1click; } });
var iotanalytics_1 = require("./iotanalytics");
Object.defineProperty(exports, "Iotanalytics", { enumerable: true, get: function () { return iotanalytics_1.Iotanalytics; } });
var iotcoredeviceadvisor_1 = require("./iotcoredeviceadvisor");
Object.defineProperty(exports, "Iotdeviceadvisor", { enumerable: true, get: function () { return iotcoredeviceadvisor_1.Iotdeviceadvisor; } });
var iotcoreforlorawan_1 = require("./iotcoreforlorawan");
Object.defineProperty(exports, "Iotwireless", { enumerable: true, get: function () { return iotcoreforlorawan_1.Iotwireless; } });
var iotdevicetester_1 = require("./iotdevicetester");
Object.defineProperty(exports, "IotDeviceTester", { enumerable: true, get: function () { return iotdevicetester_1.IotDeviceTester; } });
var iotevents_1 = require("./iotevents");
Object.defineProperty(exports, "Iotevents", { enumerable: true, get: function () { return iotevents_1.Iotevents; } });
var iotfleethubfordevicemanagement_1 = require("./iotfleethubfordevicemanagement");
Object.defineProperty(exports, "Iotfleethub", { enumerable: true, get: function () { return iotfleethubfordevicemanagement_1.Iotfleethub; } });
var iotgreengrass_1 = require("./iotgreengrass");
Object.defineProperty(exports, "Greengrass", { enumerable: true, get: function () { return iotgreengrass_1.Greengrass; } });
var iotgreengrassv2_1 = require("./iotgreengrassv2");
Object.defineProperty(exports, "GreengrassV2", { enumerable: true, get: function () { return iotgreengrassv2_1.GreengrassV2; } });
var iotsitewise_1 = require("./iotsitewise");
Object.defineProperty(exports, "Iotsitewise", { enumerable: true, get: function () { return iotsitewise_1.Iotsitewise; } });
var iotthingsgraph_1 = require("./iotthingsgraph");
Object.defineProperty(exports, "Iotthingsgraph", { enumerable: true, get: function () { return iotthingsgraph_1.Iotthingsgraph; } });
var iq_1 = require("./iq");
Object.defineProperty(exports, "Iq", { enumerable: true, get: function () { return iq_1.Iq; } });
var iqpermissions_1 = require("./iqpermissions");
Object.defineProperty(exports, "IqPermission", { enumerable: true, get: function () { return iqpermissions_1.IqPermission; } });
var keymanagementservice_1 = require("./keymanagementservice");
Object.defineProperty(exports, "Kms", { enumerable: true, get: function () { return keymanagementservice_1.Kms; } });
var lakeformation_1 = require("./lakeformation");
Object.defineProperty(exports, "Lakeformation", { enumerable: true, get: function () { return lakeformation_1.Lakeformation; } });
var lambda_1 = require("./lambda");
Object.defineProperty(exports, "Lambda", { enumerable: true, get: function () { return lambda_1.Lambda; } });
var licensemanager_1 = require("./licensemanager");
Object.defineProperty(exports, "LicenseManager", { enumerable: true, get: function () { return licensemanager_1.LicenseManager; } });
var marketplace_1 = require("./marketplace");
Object.defineProperty(exports, "AwsMarketplace", { enumerable: true, get: function () { return marketplace_1.AwsMarketplace; } });
var marketplacecatalog_1 = require("./marketplacecatalog");
Object.defineProperty(exports, "AwsMarketplaceCatalog", { enumerable: true, get: function () { return marketplacecatalog_1.AwsMarketplaceCatalog; } });
var marketplacecommerceanalyticsservice_1 = require("./marketplacecommerceanalyticsservice");
Object.defineProperty(exports, "Marketplacecommerceanalytics", { enumerable: true, get: function () { return marketplacecommerceanalyticsservice_1.Marketplacecommerceanalytics; } });
var marketplaceentitlementservice_1 = require("./marketplaceentitlementservice");
Object.defineProperty(exports, "AwsMarketplaceEntitlementService", { enumerable: true, get: function () { return marketplaceentitlementservice_1.AwsMarketplaceEntitlementService; } });
var marketplaceimagebuildingservice_1 = require("./marketplaceimagebuildingservice");
Object.defineProperty(exports, "AwsMarketplaceImageBuildingService", { enumerable: true, get: function () { return marketplaceimagebuildingservice_1.AwsMarketplaceImageBuildingService; } });
var marketplacemanagementportal_1 = require("./marketplacemanagementportal");
Object.defineProperty(exports, "AwsMarketplaceManagement", { enumerable: true, get: function () { return marketplacemanagementportal_1.AwsMarketplaceManagement; } });
var marketplacemeteringservice_1 = require("./marketplacemeteringservice");
Object.defineProperty(exports, "AwsMarketplaceMeteringService", { enumerable: true, get: function () { return marketplacemeteringservice_1.AwsMarketplaceMeteringService; } });
var marketplaceprivatemarketplace_1 = require("./marketplaceprivatemarketplace");
Object.defineProperty(exports, "AwsMarketplacePrivate", { enumerable: true, get: function () { return marketplaceprivatemarketplace_1.AwsMarketplacePrivate; } });
var marketplaceprocurementsystemsintegration_1 = require("./marketplaceprocurementsystemsintegration");
Object.defineProperty(exports, "AwsMarketplaceProcurementSystemsIntegration", { enumerable: true, get: function () { return marketplaceprocurementsystemsintegration_1.AwsMarketplaceProcurementSystemsIntegration; } });
var migrationhub_1 = require("./migrationhub");
Object.defineProperty(exports, "Mgh", { enumerable: true, get: function () { return migrationhub_1.Mgh; } });
var migrationhubstrategyrecommendations_1 = require("./migrationhubstrategyrecommendations-");
Object.defineProperty(exports, "MigrationhubStrategy", { enumerable: true, get: function () { return migrationhubstrategyrecommendations_1.MigrationhubStrategy; } });
var mobilehub_1 = require("./mobilehub");
Object.defineProperty(exports, "Mobilehub", { enumerable: true, get: function () { return mobilehub_1.Mobilehub; } });
var networkfirewall_1 = require("./networkfirewall");
Object.defineProperty(exports, "NetworkFirewall", { enumerable: true, get: function () { return networkfirewall_1.NetworkFirewall; } });
var networkmanager_1 = require("./networkmanager");
Object.defineProperty(exports, "Networkmanager", { enumerable: true, get: function () { return networkmanager_1.Networkmanager; } });
var opsworks_1 = require("./opsworks");
Object.defineProperty(exports, "Opsworks", { enumerable: true, get: function () { return opsworks_1.Opsworks; } });
var opsworksconfigurationmanagement_1 = require("./opsworksconfigurationmanagement");
Object.defineProperty(exports, "OpsworksCm", { enumerable: true, get: function () { return opsworksconfigurationmanagement_1.OpsworksCm; } });
var organizations_1 = require("./organizations");
Object.defineProperty(exports, "Organizations", { enumerable: true, get: function () { return organizations_1.Organizations; } });
var outposts_1 = require("./outposts");
Object.defineProperty(exports, "Outposts", { enumerable: true, get: function () { return outposts_1.Outposts; } });
var panorama_1 = require("./panorama");
Object.defineProperty(exports, "Panorama", { enumerable: true, get: function () { return panorama_1.Panorama; } });
var performanceinsights_1 = require("./performanceinsights");
Object.defineProperty(exports, "Pi", { enumerable: true, get: function () { return performanceinsights_1.Pi; } });
var pricelist_1 = require("./pricelist");
Object.defineProperty(exports, "Pricing", { enumerable: true, get: function () { return pricelist_1.Pricing; } });
var proton_1 = require("./proton");
Object.defineProperty(exports, "Proton", { enumerable: true, get: function () { return proton_1.Proton; } });
var purchaseordersconsole_1 = require("./purchaseordersconsole");
Object.defineProperty(exports, "PurchaseOrders", { enumerable: true, get: function () { return purchaseordersconsole_1.PurchaseOrders; } });
var resiliencehubservice_1 = require("./resiliencehubservice");
Object.defineProperty(exports, "Resiliencehub", { enumerable: true, get: function () { return resiliencehubservice_1.Resiliencehub; } });
var resourceaccessmanager_1 = require("./resourceaccessmanager");
Object.defineProperty(exports, "Ram", { enumerable: true, get: function () { return resourceaccessmanager_1.Ram; } });
var resourcegroups_1 = require("./resourcegroups");
Object.defineProperty(exports, "ResourceGroups", { enumerable: true, get: function () { return resourcegroups_1.ResourceGroups; } });
var robomaker_1 = require("./robomaker");
Object.defineProperty(exports, "Robomaker", { enumerable: true, get: function () { return robomaker_1.Robomaker; } });
var savingsplans_1 = require("./savingsplans");
Object.defineProperty(exports, "Savingsplans", { enumerable: true, get: function () { return savingsplans_1.Savingsplans; } });
var secretsmanager_1 = require("./secretsmanager");
Object.defineProperty(exports, "Secretsmanager", { enumerable: true, get: function () { return secretsmanager_1.Secretsmanager; } });
var securityhub_1 = require("./securityhub");
Object.defineProperty(exports, "Securityhub", { enumerable: true, get: function () { return securityhub_1.Securityhub; } });
var securitytokenservice_1 = require("./securitytokenservice");
Object.defineProperty(exports, "Sts", { enumerable: true, get: function () { return securitytokenservice_1.Sts; } });
var serverlessapplicationrepository_1 = require("./serverlessapplicationrepository");
Object.defineProperty(exports, "Serverlessrepo", { enumerable: true, get: function () { return serverlessapplicationrepository_1.Serverlessrepo; } });
var servermigrationservice_1 = require("./servermigrationservice");
Object.defineProperty(exports, "Sms", { enumerable: true, get: function () { return servermigrationservice_1.Sms; } });
var servicecatalog_1 = require("./servicecatalog");
Object.defineProperty(exports, "Servicecatalog", { enumerable: true, get: function () { return servicecatalog_1.Servicecatalog; } });
var shield_1 = require("./shield");
Object.defineProperty(exports, "Shield", { enumerable: true, get: function () { return shield_1.Shield; } });
var signer_1 = require("./signer");
Object.defineProperty(exports, "Signer", { enumerable: true, get: function () { return signer_1.Signer; } });
var snowball_1 = require("./snowball");
Object.defineProperty(exports, "Snowball", { enumerable: true, get: function () { return snowball_1.Snowball; } });
var snowdevicemanagement_1 = require("./snowdevicemanagement");
Object.defineProperty(exports, "SnowDeviceManagement", { enumerable: true, get: function () { return snowdevicemanagement_1.SnowDeviceManagement; } });
var sqlworkbench_1 = require("./sqlworkbench");
Object.defineProperty(exports, "Sqlworkbench", { enumerable: true, get: function () { return sqlworkbench_1.Sqlworkbench; } });
var sso_1 = require("./sso");
Object.defineProperty(exports, "Sso", { enumerable: true, get: function () { return sso_1.Sso; } });
var ssodirectory_1 = require("./ssodirectory");
Object.defineProperty(exports, "SsoDirectory", { enumerable: true, get: function () { return ssodirectory_1.SsoDirectory; } });
var stepfunctions_1 = require("./stepfunctions");
Object.defineProperty(exports, "States", { enumerable: true, get: function () { return stepfunctions_1.States; } });
var support_1 = require("./support");
Object.defineProperty(exports, "Support", { enumerable: true, get: function () { return support_1.Support; } });
var systemsmanager_1 = require("./systemsmanager");
Object.defineProperty(exports, "Ssm", { enumerable: true, get: function () { return systemsmanager_1.Ssm; } });
var systemsmanagerguiconnect_1 = require("./systemsmanagerguiconnect");
Object.defineProperty(exports, "SsmGuiconnect", { enumerable: true, get: function () { return systemsmanagerguiconnect_1.SsmGuiconnect; } });
var systemsmanagerincidentmanager_1 = require("./systemsmanagerincidentmanager");
Object.defineProperty(exports, "SsmIncidents", { enumerable: true, get: function () { return systemsmanagerincidentmanager_1.SsmIncidents; } });
var systemsmanagerincidentmanagercontacts_1 = require("./systemsmanagerincidentmanagercontacts");
Object.defineProperty(exports, "SsmContacts", { enumerable: true, get: function () { return systemsmanagerincidentmanagercontacts_1.SsmContacts; } });
var tageditor_1 = require("./tageditor");
Object.defineProperty(exports, "ResourceExplorer", { enumerable: true, get: function () { return tageditor_1.ResourceExplorer; } });
var tiros_1 = require("./tiros");
Object.defineProperty(exports, "Tiros", { enumerable: true, get: function () { return tiros_1.Tiros; } });
var transferfamily_1 = require("./transferfamily");
Object.defineProperty(exports, "Transfer", { enumerable: true, get: function () { return transferfamily_1.Transfer; } });
var trustedadvisor_1 = require("./trustedadvisor");
Object.defineProperty(exports, "Trustedadvisor", { enumerable: true, get: function () { return trustedadvisor_1.Trustedadvisor; } });
var waf_1 = require("./waf");
Object.defineProperty(exports, "Waf", { enumerable: true, get: function () { return waf_1.Waf; } });
var wafregional_1 = require("./wafregional");
Object.defineProperty(exports, "WafRegional", { enumerable: true, get: function () { return wafregional_1.WafRegional; } });
var wafv2_1 = require("./wafv2");
Object.defineProperty(exports, "Wafv2", { enumerable: true, get: function () { return wafv2_1.Wafv2; } });
var well_architectedtool_1 = require("./well-architectedtool");
Object.defineProperty(exports, "Wellarchitected", { enumerable: true, get: function () { return well_architectedtool_1.Wellarchitected; } });
var x_ray_1 = require("./x-ray");
Object.defineProperty(exports, "Xray", { enumerable: true, get: function () { return x_ray_1.Xray; } });
var cloudwatchapplicationinsights_1 = require("./cloudwatchapplicationinsights");
Object.defineProperty(exports, "Applicationinsights", { enumerable: true, get: function () { return cloudwatchapplicationinsights_1.Applicationinsights; } });
var databasequerymetadataservice_1 = require("./databasequerymetadataservice");
Object.defineProperty(exports, "Dbqms", { enumerable: true, get: function () { return databasequerymetadataservice_1.Dbqms; } });
var datapipeline_1 = require("./datapipeline");
Object.defineProperty(exports, "Datapipeline", { enumerable: true, get: function () { return datapipeline_1.Datapipeline; } });
var elasticloadbalancing_1 = require("./elasticloadbalancing");
Object.defineProperty(exports, "Elasticloadbalancing", { enumerable: true, get: function () { return elasticloadbalancing_1.Elasticloadbalancing; } });
var elasticloadbalancingv2_1 = require("./elasticloadbalancingv2");
Object.defineProperty(exports, "ElasticloadbalancingV2", { enumerable: true, get: function () { return elasticloadbalancingv2_1.ElasticloadbalancingV2; } });
var elementalsupportcases_1 = require("./elementalsupportcases");
Object.defineProperty(exports, "ElementalSupportCases", { enumerable: true, get: function () { return elementalsupportcases_1.ElementalSupportCases; } });
var elementalsupportcontent_1 = require("./elementalsupportcontent");
Object.defineProperty(exports, "ElementalSupportContent", { enumerable: true, get: function () { return elementalsupportcontent_1.ElementalSupportContent; } });
var high_volumeoutboundcommunications_1 = require("./high-volumeoutboundcommunications");
Object.defineProperty(exports, "ConnectCampaigns", { enumerable: true, get: function () { return high_volumeoutboundcommunications_1.ConnectCampaigns; } });
var identityandaccessmanagement_1 = require("./identityandaccessmanagement");
Object.defineProperty(exports, "Iam", { enumerable: true, get: function () { return identityandaccessmanagement_1.Iam; } });
var launchwizard_1 = require("./launchwizard");
Object.defineProperty(exports, "Launchwizard", { enumerable: true, get: function () { return launchwizard_1.Launchwizard; } });
var servicequotas_1 = require("./servicequotas");
Object.defineProperty(exports, "Servicequotas", { enumerable: true, get: function () { return servicequotas_1.Servicequotas; } });
