// cdk-iam-floyd/lib/generated/marketplacecommerceanalyticsservice.js
"use strict";
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
exports.Marketplacecommerceanalytics = void 0;
const JSII_RTTI_SYMBOL_1 = Symbol.for("jsii.rtti");
const shared_1 = require("../shared");
/**
 * (experimental) Statement provider for service [marketplacecommerceanalytics](https://docs.aws.amazon.com/service-authorization/latest/reference/list_awsmarketplacecommerceanalyticsservice.html).
 *
 * @experimental
 */
class Marketplacecommerceanalytics extends shared_1.PolicyStatement {
    /**
     * (experimental) Statement provider for service [marketplacecommerceanalytics](https://docs.aws.amazon.com/service-authorization/latest/reference/list_awsmarketplacecommerceanalyticsservice.html).
     *
     * @experimental
     */
    constructor(props) {
        super(props);
        /**
         * (experimental) Holds the prefix of the service actions, e.g. `ec2`.
         *
         * @experimental
         */
        this.servicePrefix = 'marketplacecommerceanalytics';
        /**
         * @experimental
         */
        this.accessLevelList = {
            Write: [
                'GenerateDataSet',
                'StartSupportDataExport'
            ]
        };
    }
    /**
     * (experimental) Request a data set to be published to your Amazon S3 bucket.
     *
     * Access Level: Write
     *
     * @experimental
     */
    toGenerateDataSet() {
        return this.to('GenerateDataSet');
    }
    /**
     * (experimental) Request a support data set to be published to your Amazon S3 bucket.
     *
     * Access Level: Write
     *
     * @experimental
     */
    toStartSupportDataExport() {
        return this.to('StartSupportDataExport');
    }
}
exports.Marketplacecommerceanalytics = Marketplacecommerceanalytics;
_a = JSII_RTTI_SYMBOL_1;
Marketplacecommerceanalytics[_a] = { fqn: "cdk-iam-floyd.Marketplacecommerceanalytics", version: "0.300.0" };
