// cdk-iam-floyd/lib/generated/migrationhubstrategyrecommendations-.js
"use strict";
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
exports.MigrationhubStrategy = void 0;
const JSII_RTTI_SYMBOL_1 = Symbol.for("jsii.rtti");
const shared_1 = require("../shared");
/**
 * (experimental) Statement provider for service [migrationhub-strategy](https://docs.aws.amazon.com/service-authorization/latest/reference/list_awsmigrationhubstrategyrecommendations..html).
 *
 * @experimental
 */
class MigrationhubStrategy extends shared_1.PolicyStatement {
    /**
     * (experimental) Statement provider for service [migrationhub-strategy](https://docs.aws.amazon.com/service-authorization/latest/reference/list_awsmigrationhubstrategyrecommendations..html).
     *
     * @experimental
     */
    constructor(props) {
        super(props);
        /**
         * (experimental) Holds the prefix of the service actions, e.g. `ec2`.
         *
         * @experimental
         */
        this.servicePrefix = 'migrationhub-strategy';
        /**
         * @experimental
         */
        this.accessLevelList = {
            Read: [
                'GetAntiPattern',
                'GetApplicationComponentDetails',
                'GetApplicationComponentStrategies',
                'GetAssessment',
                'GetImportFileTask',
                'GetMessage',
                'GetPortfolioPreferences',
                'GetPortfolioSummary',
                'GetRecommendationReportDetails',
                'GetServerDetails',
                'GetServerStrategies'
            ],
            List: [
                'ListAntiPatterns',
                'ListApplicationComponents',
                'ListCollectors',
                'ListImportFileTask',
                'ListJarArtifacts',
                'ListServers'
            ],
            Write: [
                'PutPortfolioPreferences',
                'RegisterCollector',
                'SendMessage',
                'StartAssessment',
                'StartImportFileTask',
                'StartRecommendationReportGeneration',
                'StopAssessment',
                'UpdateApplicationComponentConfig',
                'UpdateServerConfig'
            ]
        };
    }
    /**
     * (experimental) Grants permission to get details of each anti pattern that collector should look at in a customer's environment.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/migrationhub-strategy/latest/APIReference/API_GetAntiPattern.html
     *
     * @experimental
     */
    toGetAntiPattern() {
        return this.to('GetAntiPattern');
    }
    /**
     * (experimental) Grants permission to get details of an application.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/migrationhub-strategy/latest/APIReference/API_GetApplicationComponentDetails.html
     *
     * @experimental
     */
    toGetApplicationComponentDetails() {
        return this.to('GetApplicationComponentDetails');
    }
    /**
     * (experimental) Grants permission to get a list of all recommended strategies and tools for an application running in a server.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/migrationhub-strategy/latest/APIReference/API_GetApplicationComponentStrategies.html
     *
     * @experimental
     */
    toGetApplicationComponentStrategies() {
        return this.to('GetApplicationComponentStrategies');
    }
    /**
     * (experimental) Grants permission to retrieve status of an on-going assessment.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/migrationhub-strategy/latest/APIReference/API_GetAssessment.html
     *
     * @experimental
     */
    toGetAssessment() {
        return this.to('GetAssessment');
    }
    /**
     * (experimental) Grants permission to get details of a specific import task.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/migrationhub-strategy/latest/APIReference/API_GetImportFileTask.html
     *
     * @experimental
     */
    toGetImportFileTask() {
        return this.to('GetImportFileTask');
    }
    /**
     * (experimental) Grants permission to the collector to receive information from the service.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/migrationhub-strategy/latest/APIReference/API_GetMessage.html
     *
     * @experimental
     */
    toGetMessage() {
        return this.to('GetMessage');
    }
    /**
     * (experimental) Grants permission to retrieve customer migration/Modernization preferences.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/migrationhub-strategy/latest/APIReference/API_GetPortfolioPreferences.html
     *
     * @experimental
     */
    toGetPortfolioPreferences() {
        return this.to('GetPortfolioPreferences');
    }
    /**
     * (experimental) Grants permission to retrieve overall summary (number-of servers to rehost etc as well as overall number of anti patterns).
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/migrationhub-strategy/latest/APIReference/API_GetPortfolioSummary.html
     *
     * @experimental
     */
    toGetPortfolioSummary() {
        return this.to('GetPortfolioSummary');
    }
    /**
     * (experimental) Grants permission to retrieve detailed information about a recommendation report.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/migrationhub-strategy/latest/APIReference/API_GetRecommendationReportDetails.html
     *
     * @experimental
     */
    toGetRecommendationReportDetails() {
        return this.to('GetRecommendationReportDetails');
    }
    /**
     * (experimental) Grants permission to get info about a specific server.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/migrationhub-strategy/latest/APIReference/API_GetServerDetails.html
     *
     * @experimental
     */
    toGetServerDetails() {
        return this.to('GetServerDetails');
    }
    /**
     * (experimental) Grants permission to get recommended strategies and tools for a specific server.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/migrationhub-strategy/latest/APIReference/API_GetServerStrategies.html
     *
     * @experimental
     */
    toGetServerStrategies() {
        return this.to('GetServerStrategies');
    }
    /**
     * (experimental) Grants permission to get a list of all anti patterns that collector should look for in a customer's environment.
     *
     * Access Level: List
     *
     * https://docs.aws.amazon.com/migrationhub-strategy/latest/APIReference/API_ListAntiPatterns.html
     *
     * @experimental
     */
    toListAntiPatterns() {
        return this.to('ListAntiPatterns');
    }
    /**
     * (experimental) Grants permission to get a list of all applications running on servers on customer's servers.
     *
     * Access Level: List
     *
     * https://docs.aws.amazon.com/migrationhub-strategy/latest/APIReference/API_ListApplicationComponents.html
     *
     * @experimental
     */
    toListApplicationComponents() {
        return this.to('ListApplicationComponents');
    }
    /**
     * (experimental) Grants permission to get a list of all collectors installed by the customer.
     *
     * Access Level: List
     *
     * https://docs.aws.amazon.com/migrationhub-strategy/latest/APIReference/API_ListCollectors.html
     *
     * @experimental
     */
    toListCollectors() {
        return this.to('ListCollectors');
    }
    /**
     * (experimental) Grants permission to get list of all imports performed by the customer.
     *
     * Access Level: List
     *
     * https://docs.aws.amazon.com/migrationhub-strategy/latest/APIReference/API_ListImportFileTask.html
     *
     * @experimental
     */
    toListImportFileTask() {
        return this.to('ListImportFileTask');
    }
    /**
     * (experimental) Grants permission to get a list of binaries that collector should assess.
     *
     * Access Level: List
     *
     * https://docs.aws.amazon.com/migrationhub-strategy/latest/APIReference/API_ListJarArtifacts.html
     *
     * @experimental
     */
    toListJarArtifacts() {
        return this.to('ListJarArtifacts');
    }
    /**
     * (experimental) Grants permission to get a list of all servers in a customer's environment.
     *
     * Access Level: List
     *
     * https://docs.aws.amazon.com/migrationhub-strategy/latest/APIReference/API_ListServers.html
     *
     * @experimental
     */
    toListServers() {
        return this.to('ListServers');
    }
    /**
     * (experimental) Grants permission to save customer's Migration/Modernization preferences.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/migrationhub-strategy/latest/APIReference/API_PutPortfolioPreferences.html
     *
     * @experimental
     */
    toPutPortfolioPreferences() {
        return this.to('PutPortfolioPreferences');
    }
    /**
     * (experimental) Grants permission to register the collector to receive an ID and to start receiving messages from the service.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/migrationhub-strategy/latest/APIReference/API_RegisterCollector.html
     *
     * @experimental
     */
    toRegisterCollector() {
        return this.to('RegisterCollector');
    }
    /**
     * (experimental) Grants permission to the collector to send information to the service.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/migrationhub-strategy/latest/APIReference/API_SendMessage.html
     *
     * @experimental
     */
    toSendMessage() {
        return this.to('SendMessage');
    }
    /**
     * (experimental) Grants permission to start assessment in a customer's environment (collect data from all servers and provide recommendations).
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/migrationhub-strategy/latest/APIReference/API_StartAssessment.html
     *
     * @experimental
     */
    toStartAssessment() {
        return this.to('StartAssessment');
    }
    /**
     * (experimental) Grants permission to start importing data from a file provided by customer.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/migrationhub-strategy/latest/APIReference/API_StartImportFileTask.html
     *
     * @experimental
     */
    toStartImportFileTask() {
        return this.to('StartImportFileTask');
    }
    /**
     * (experimental) Grants permission to start generating a recommendation report.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/migrationhub-strategy/latest/APIReference/API_StartRecommendationReportGeneration.html
     *
     * @experimental
     */
    toStartRecommendationReportGeneration() {
        return this.to('StartRecommendationReportGeneration');
    }
    /**
     * (experimental) Grants permission to stop an on-going assessment.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/migrationhub-strategy/latest/APIReference/API_StopAssessment.html
     *
     * @experimental
     */
    toStopAssessment() {
        return this.to('StopAssessment');
    }
    /**
     * (experimental) Grants permission to update details for an application.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/migrationhub-strategy/latest/APIReference/API_UpdateApplicationComponentConfig.html
     *
     * @experimental
     */
    toUpdateApplicationComponentConfig() {
        return this.to('UpdateApplicationComponentConfig');
    }
    /**
     * (experimental) Grants permission to update info on a server along with the recommended strategy.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/migrationhub-strategy/latest/APIReference/API_UpdateServerConfig.html
     *
     * @experimental
     */
    toUpdateServerConfig() {
        return this.to('UpdateServerConfig');
    }
}
exports.MigrationhubStrategy = MigrationhubStrategy;
_a = JSII_RTTI_SYMBOL_1;
MigrationhubStrategy[_a] = { fqn: "cdk-iam-floyd.MigrationhubStrategy", version: "0.300.0" };
