// cdk-iam-floyd/lib/generated/mobileanalytics.js
"use strict";
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
exports.Mobileanalytics = void 0;
const JSII_RTTI_SYMBOL_1 = Symbol.for("jsii.rtti");
const shared_1 = require("../shared");
/**
 * (experimental) Statement provider for service [mobileanalytics](https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazonmobileanalytics.html).
 *
 * @experimental
 */
class Mobileanalytics extends shared_1.PolicyStatement {
    /**
     * (experimental) Statement provider for service [mobileanalytics](https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazonmobileanalytics.html).
     *
     * @experimental
     */
    constructor(props) {
        super(props);
        /**
         * (experimental) Holds the prefix of the service actions, e.g. `ec2`.
         *
         * @experimental
         */
        this.servicePrefix = 'mobileanalytics';
        /**
         * @experimental
         */
        this.accessLevelList = {
            Read: [
                'GetFinancialReports',
                'GetReports'
            ],
            Write: [
                'PutEvents'
            ]
        };
    }
    /**
     * (experimental) Grant access to financial metrics for an app.
     *
     * Access Level: Read
     *
     * @experimental
     */
    toGetFinancialReports() {
        return this.to('GetFinancialReports');
    }
    /**
     * (experimental) Grant access to standard metrics for an app.
     *
     * Access Level: Read
     *
     * @experimental
     */
    toGetReports() {
        return this.to('GetReports');
    }
    /**
     * (experimental) The PutEvents operation records one or more events.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/mobileanalytics/latest/ug/PutEvents.html
     *
     * @experimental
     */
    toPutEvents() {
        return this.to('PutEvents');
    }
}
exports.Mobileanalytics = Mobileanalytics;
_a = JSII_RTTI_SYMBOL_1;
Mobileanalytics[_a] = { fqn: "cdk-iam-floyd.Mobileanalytics", version: "0.300.0" };
