// cdk-iam-floyd/lib/generated/tiros.js
"use strict";
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
exports.Tiros = void 0;
const JSII_RTTI_SYMBOL_1 = Symbol.for("jsii.rtti");
const shared_1 = require("../shared");
/**
 * (experimental) Statement provider for service [tiros](https://docs.aws.amazon.com/service-authorization/latest/reference/list_awstiros.html).
 *
 * @experimental
 */
class Tiros extends shared_1.PolicyStatement {
    /**
     * (experimental) Statement provider for service [tiros](https://docs.aws.amazon.com/service-authorization/latest/reference/list_awstiros.html).
     *
     * @experimental
     */
    constructor(props) {
        super(props);
        /**
         * (experimental) Holds the prefix of the service actions, e.g. `ec2`.
         *
         * @experimental
         */
        this.servicePrefix = 'tiros';
        /**
         * @experimental
         */
        this.accessLevelList = {
            Write: [
                'CreateQuery'
            ],
            Read: [
                'GetQueryAnswer',
                'GetQueryExplanation'
            ]
        };
    }
    /**
     * (experimental) Grants permission to create a VPC reachability query.
     *
     * Access Level: Write
     *
     * https://docs.aws.amazon.com/vpc/latest/reachability/security_iam_required-API-permissions.html
     *
     * @experimental
     */
    toCreateQuery() {
        return this.to('CreateQuery');
    }
    /**
     * (experimental) Grants permission to get VPC reachability query answers.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/vpc/latest/reachability/security_iam_required-API-permissions.html
     *
     * @experimental
     */
    toGetQueryAnswer() {
        return this.to('GetQueryAnswer');
    }
    /**
     * (experimental) Grants permission to get VPC reachability query explanations.
     *
     * Access Level: Read
     *
     * https://docs.aws.amazon.com/vpc/latest/reachability/security_iam_required-API-permissions.html
     *
     * @experimental
     */
    toGetQueryExplanation() {
        return this.to('GetQueryExplanation');
    }
}
exports.Tiros = Tiros;
_a = JSII_RTTI_SYMBOL_1;
Tiros[_a] = { fqn: "cdk-iam-floyd.Tiros", version: "0.300.0" };
